
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../01-Metabarcoding%20ampliseq%2012S%20Riaz/">
      
      
        <link rel="next" href="../03-data_quality/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.37">
    
    
      
        <title>02-dfprep-12S - GMGI Fisheries Handbook</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.8c3ca2c6.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="blue" data-md-color-accent="blue">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#datatable-preparation-base-script-for-edna-metabarcoding" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="GMGI Fisheries Handbook" class="md-header__button md-logo" aria-label="GMGI Fisheries Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            GMGI Fisheries Handbook
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              02-dfprep-12S
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="blue" data-md-color-accent="blue"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6m0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4M7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="blue" data-md-color-accent="blue"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/GMGI-Fisheries/resources" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="GMGI Fisheries Handbook" class="md-nav__button md-logo" aria-label="GMGI Fisheries Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    GMGI Fisheries Handbook
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/GMGI-Fisheries/resources" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Computing Hardware
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Computing Hardware
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Computing_GMGI/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    GMGI
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Computing_NU/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Northeastern University
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Data_Management/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Data Management
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Data_To_Server/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Data Transfer
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Computing Resources
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Computing Resources
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Computing_Begin_Resources/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Basics
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Data%20Visualization%20in%20R/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    R Resources
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tbd.md" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Intro to Github
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Introduction%20to%20HPC/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Intro to HPC
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../.md" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Intro to Conda
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Introduction%20to%20Containers/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Intro to Containers
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../Introduction%20to%20Workflow%20Managers/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Intro to Workflow Managers
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" checked>
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    eDNA: 12S metabarcoding
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            eDNA: 12S metabarcoding
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../00-creating_eDNA_conda_env/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    00-Create-conda-env
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../01-Metabarcoding%20ampliseq%2012S%20Riaz/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    01-ampliseq-12S
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    02-dfprep-12S
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    02-dfprep-12S
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#load-libraries" class="md-nav__link">
    <span class="md-ellipsis">
      Load libraries
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#metadata-input" class="md-nav__link">
    <span class="md-ellipsis">
      Metadata input
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Metadata input">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#identify-paths-for-metadata-and-project-data" class="md-nav__link">
    <span class="md-ellipsis">
      Identify paths for metadata and project data
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#load-project-metadata" class="md-nav__link">
    <span class="md-ellipsis">
      Load project metadata
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#load-database-metadata" class="md-nav__link">
    <span class="md-ellipsis">
      Load database metadata
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#blast-data-input" class="md-nav__link">
    <span class="md-ellipsis">
      BLAST data input
    </span>
  </a>
  
    <nav class="md-nav" aria-label="BLAST data input">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gmgi-database" class="md-nav__link">
    <span class="md-ellipsis">
      GMGI database
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mitofish-database" class="md-nav__link">
    <span class="md-ellipsis">
      Mitofish database
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ncbi-database" class="md-nav__link">
    <span class="md-ellipsis">
      NCBI database
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#load-asv-sequences" class="md-nav__link">
    <span class="md-ellipsis">
      Load ASV sequences
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#load-dada2-asv-table" class="md-nav__link">
    <span class="md-ellipsis">
      Load DADA2 ASV Table
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#taxonomic-assignment" class="md-nav__link">
    <span class="md-ellipsis">
      Taxonomic Assignment
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Taxonomic Assignment">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#identify-any-asvs-that-contain-multiple-hits-within-the-gmgi-database" class="md-nav__link">
    <span class="md-ellipsis">
      Identify any ASVs that contain multiple hits within the GMGI database
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Identify any ASVs that contain multiple hits within the GMGI database">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#create-list-of-those-asvs-with-multiple-hits" class="md-nav__link">
    <span class="md-ellipsis">
      Create list of those ASVs with multiple hits
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#choosing-one-of-several-hits" class="md-nav__link">
    <span class="md-ellipsis">
      Choosing one of several hits.
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#confirming-that-all-entries-have-been-dealth-with" class="md-nav__link">
    <span class="md-ellipsis">
      Confirming that all entries have been dealth with
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#identify-entries-that-mismatch-between-gmgi-mitofish-and-ncbi-databases" class="md-nav__link">
    <span class="md-ellipsis">
      Identify entries that mismatch between GMGI, Mitofish, and NCBI databases
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assign-taxonomy-based-on-hierarchical-approach" class="md-nav__link">
    <span class="md-ellipsis">
      Assign taxonomy based on hierarchical approach
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#edit-taxonomy-annotations-based-on-mismatch-table" class="md-nav__link">
    <span class="md-ellipsis">
      Edit taxonomy annotations based on mismatch table
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Edit taxonomy annotations based on mismatch table">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#confirm-all-entries-are-dealt-with" class="md-nav__link">
    <span class="md-ellipsis">
      Confirm all entries are dealt with
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#adjusting-common-name-for-those-entries-that-dont-have-one-from-mito-or-ncbi" class="md-nav__link">
    <span class="md-ellipsis">
      Adjusting common name for those entries that don’t have one (from Mito or NCBI)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#filtering" class="md-nav__link">
    <span class="md-ellipsis">
      Filtering
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Filtering">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#filter-out-reads-that-are-less-than-10-in-each-sample-per-asv" class="md-nav__link">
    <span class="md-ellipsis">
      Filter out reads that are less than 10 in each sample per ASV
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filter-out-reads-that-are-less-than-a-certain-of-asv-row-total-per-sample" class="md-nav__link">
    <span class="md-ellipsis">
      Filter out reads that are less than a certain % of ASV (row) total per sample
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filter-out-samples-with-less-than-x-number-of-reads" class="md-nav__link">
    <span class="md-ellipsis">
      Filter out samples with less than X number of reads
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#create-asv-breakdown-table" class="md-nav__link">
    <span class="md-ellipsis">
      Create ASV breakdown table
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#collapsing-read-counts-by-species-name" class="md-nav__link">
    <span class="md-ellipsis">
      Collapsing read counts by species name
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#creating-results-output" class="md-nav__link">
    <span class="md-ellipsis">
      Creating results output
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../03-data_quality/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    03-data-quality
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../04-relative_abundance_heatmap/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    04-relative-abund-heatmap
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../05-biodiversity/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    05-biodiversity
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    eDNA: COI metabarcoding
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            eDNA: COI metabarcoding
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../00-creating_eDNA_conda_env/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    00-Create-conda-env
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../eDNA%20COI%20metab/01-Metabarcoding%20ampliseq%20COI/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    01-ampliseq-COI
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_8" >
        
          
          <label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    eDNA: qPCR
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_8">
            <span class="md-nav__icon md-icon"></span>
            eDNA: qPCR
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../eDNA%20qPCR/qPCR_analysis/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    01-dfprep-analysis
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_9" >
        
          
          <label class="md-nav__link" for="__nav_9" id="__nav_9_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Population Genomics
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_9_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_9">
            <span class="md-nav__icon md-icon"></span>
            Population Genomics
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../PopGen/eDNA_01/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    coming soon!
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_10" >
        
          
          <label class="md-nav__link" for="__nav_10" id="__nav_10_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Epigenetic Aging
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_10_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_10">
            <span class="md-nav__icon md-icon"></span>
            Epigenetic Aging
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../EpiAge/eDNA_01.md" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    coming soon!
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_11" >
        
          
          <label class="md-nav__link" for="__nav_11" id="__nav_11_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Genome Assembly
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_11_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_11">
            <span class="md-nav__icon md-icon"></span>
            Genome Assembly
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../tbd.md" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    coming soon!
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#load-libraries" class="md-nav__link">
    <span class="md-ellipsis">
      Load libraries
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#metadata-input" class="md-nav__link">
    <span class="md-ellipsis">
      Metadata input
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Metadata input">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#identify-paths-for-metadata-and-project-data" class="md-nav__link">
    <span class="md-ellipsis">
      Identify paths for metadata and project data
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#load-project-metadata" class="md-nav__link">
    <span class="md-ellipsis">
      Load project metadata
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#load-database-metadata" class="md-nav__link">
    <span class="md-ellipsis">
      Load database metadata
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#blast-data-input" class="md-nav__link">
    <span class="md-ellipsis">
      BLAST data input
    </span>
  </a>
  
    <nav class="md-nav" aria-label="BLAST data input">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gmgi-database" class="md-nav__link">
    <span class="md-ellipsis">
      GMGI database
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mitofish-database" class="md-nav__link">
    <span class="md-ellipsis">
      Mitofish database
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ncbi-database" class="md-nav__link">
    <span class="md-ellipsis">
      NCBI database
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#load-asv-sequences" class="md-nav__link">
    <span class="md-ellipsis">
      Load ASV sequences
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#load-dada2-asv-table" class="md-nav__link">
    <span class="md-ellipsis">
      Load DADA2 ASV Table
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#taxonomic-assignment" class="md-nav__link">
    <span class="md-ellipsis">
      Taxonomic Assignment
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Taxonomic Assignment">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#identify-any-asvs-that-contain-multiple-hits-within-the-gmgi-database" class="md-nav__link">
    <span class="md-ellipsis">
      Identify any ASVs that contain multiple hits within the GMGI database
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Identify any ASVs that contain multiple hits within the GMGI database">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#create-list-of-those-asvs-with-multiple-hits" class="md-nav__link">
    <span class="md-ellipsis">
      Create list of those ASVs with multiple hits
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#choosing-one-of-several-hits" class="md-nav__link">
    <span class="md-ellipsis">
      Choosing one of several hits.
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#confirming-that-all-entries-have-been-dealth-with" class="md-nav__link">
    <span class="md-ellipsis">
      Confirming that all entries have been dealth with
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#identify-entries-that-mismatch-between-gmgi-mitofish-and-ncbi-databases" class="md-nav__link">
    <span class="md-ellipsis">
      Identify entries that mismatch between GMGI, Mitofish, and NCBI databases
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assign-taxonomy-based-on-hierarchical-approach" class="md-nav__link">
    <span class="md-ellipsis">
      Assign taxonomy based on hierarchical approach
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#edit-taxonomy-annotations-based-on-mismatch-table" class="md-nav__link">
    <span class="md-ellipsis">
      Edit taxonomy annotations based on mismatch table
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Edit taxonomy annotations based on mismatch table">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#confirm-all-entries-are-dealt-with" class="md-nav__link">
    <span class="md-ellipsis">
      Confirm all entries are dealt with
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#adjusting-common-name-for-those-entries-that-dont-have-one-from-mito-or-ncbi" class="md-nav__link">
    <span class="md-ellipsis">
      Adjusting common name for those entries that don’t have one (from Mito or NCBI)
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#filtering" class="md-nav__link">
    <span class="md-ellipsis">
      Filtering
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Filtering">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#filter-out-reads-that-are-less-than-10-in-each-sample-per-asv" class="md-nav__link">
    <span class="md-ellipsis">
      Filter out reads that are less than 10 in each sample per ASV
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filter-out-reads-that-are-less-than-a-certain-of-asv-row-total-per-sample" class="md-nav__link">
    <span class="md-ellipsis">
      Filter out reads that are less than a certain % of ASV (row) total per sample
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filter-out-samples-with-less-than-x-number-of-reads" class="md-nav__link">
    <span class="md-ellipsis">
      Filter out samples with less than X number of reads
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#create-asv-breakdown-table" class="md-nav__link">
    <span class="md-ellipsis">
      Create ASV breakdown table
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#collapsing-read-counts-by-species-name" class="md-nav__link">
    <span class="md-ellipsis">
      Collapsing read counts by species name
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#creating-results-output" class="md-nav__link">
    <span class="md-ellipsis">
      Creating results output
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
    <a href="https://github.com/GMGI-Fisheries/resources/blob/master/docs/eDNA 12S metab/02-datatable_prep_12S_Riaz.md" title="Edit this page" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75z"/></svg>
    </a>
  
  


<h1 id="datatable-preparation-base-script-for-edna-metabarcoding">Datatable preparation base script for eDNA metabarcoding</h1>
<p><strong>.Rmd script</strong></p>
<p>Example data from <a href="https://onlinelibrary.wiley.com/doi/full/10.1002/edn3.70138">Polinski et al.,
2025</a></p>
<p>This script takes your Blast output from the GMGI database, Mitofish
database, and NCBI database to create one datatable with read counts and
taxonomic assignment.</p>
<p><strong>Workflow summary:</strong><br />
1. Load libraries<br />
2. Load metadata<br />
3. Load BLAST output from GMGI, Mitofish, and NCBI<br />
4. Load DADA2 ASV Table<br />
5. Taxonomic Assignment<br />
- 5a. Identify ASVs with multiple hits from GMGI’s database<br />
- 5b. Identify entries that mismatch between GMGI, Mitofish, and NCBI
databases<br />
- 5c. Assign taxonomy based on hierarchical approach<br />
- 5d. Edit taxonomy annotations based on mismatch table choices<br />
- 5e. Adjusting common name and category for those entries that don’t
have one (from Mito or NCBI)<br />
6. Filtering: Filter ASV by less than 0.1% reads and then collapse by
group<br />
7. Collapsing read counts by species name<br />
8. Creating results output</p>
<h2 id="load-libraries">Load libraries</h2>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span><span class="w"> </span><span class="c1">## for plotting</span>
<span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span><span class="w"> </span><span class="c1">## for data table manipulation</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## 
## Attaching package: &#39;dplyr&#39;

## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag

## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">tidyr</span><span class="p">)</span><span class="w"> </span><span class="c1">## for data table manipulation</span>
<span class="nf">library</span><span class="p">(</span><span class="n">readr</span><span class="p">)</span><span class="w"> </span><span class="c1">## for reading in tsv files</span>
<span class="nf">library</span><span class="p">(</span><span class="n">readxl</span><span class="p">)</span><span class="w"> </span><span class="c1">## for reading in excel files</span>
<span class="nf">library</span><span class="p">(</span><span class="n">stringr</span><span class="p">)</span><span class="w"> </span><span class="c1">## for data transformation</span>
<span class="nf">library</span><span class="p">(</span><span class="n">strex</span><span class="p">)</span><span class="w"> </span><span class="c1">## for data transformation</span>
<span class="nf">library</span><span class="p">(</span><span class="n">writexl</span><span class="p">)</span><span class="w"> </span><span class="c1">## for excel output</span>
<span class="nf">library</span><span class="p">(</span><span class="n">purrr</span><span class="p">)</span><span class="w"> </span><span class="c1">## for data transformation</span>
<span class="nf">library</span><span class="p">(</span><span class="n">funrar</span><span class="p">)</span><span class="w"> </span><span class="c1">## for make_relative()</span>
<span class="nf">library</span><span class="p">(</span><span class="n">tidyverse</span><span class="p">)</span><span class="w"> </span><span class="c1">## for data table manipulation</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
## ✔ forcats   1.0.0     ✔ tibble    3.3.0
## ✔ lubridate 1.9.4

## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()
## ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">Biostrings</span><span class="p">)</span><span class="w"> </span><span class="c1">## for reading in fasta file </span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## Loading required package: BiocGenerics
## 
## Attaching package: &#39;BiocGenerics&#39;
## 
## The following objects are masked from &#39;package:lubridate&#39;:
## 
##     intersect, setdiff, union
## 
## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     combine, intersect, setdiff, union
## 
## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs
## 
## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,
##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
##     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
##     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
##     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,
##     table, tapply, union, unique, unsplit, which.max, which.min
## 
## Loading required package: S4Vectors
## Loading required package: stats4
## 
## Attaching package: &#39;S4Vectors&#39;
## 
## The following objects are masked from &#39;package:lubridate&#39;:
## 
##     second, second&lt;-
## 
## The following object is masked from &#39;package:tidyr&#39;:
## 
##     expand
## 
## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     first, rename
## 
## The following object is masked from &#39;package:utils&#39;:
## 
##     findMatches
## 
## The following objects are masked from &#39;package:base&#39;:
## 
##     expand.grid, I, unname
## 
## Loading required package: IRanges
## 
## Attaching package: &#39;IRanges&#39;
## 
## The following object is masked from &#39;package:lubridate&#39;:
## 
##     %within%
## 
## The following object is masked from &#39;package:purrr&#39;:
## 
##     reduce
## 
## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     collapse, desc, slice
## 
## The following object is masked from &#39;package:grDevices&#39;:
## 
##     windows
## 
## Loading required package: XVector
## 
## Attaching package: &#39;XVector&#39;
## 
## The following object is masked from &#39;package:purrr&#39;:
## 
##     compact
## 
## Loading required package: GenomeInfoDb
## 
## Attaching package: &#39;Biostrings&#39;
## 
## The following object is masked from &#39;package:base&#39;:
## 
##     strsplit
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">data.table</span><span class="p">)</span><span class="w">  </span><span class="c1">## for data table manipulation</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## 
## Attaching package: &#39;data.table&#39;
## 
## The following object is masked from &#39;package:IRanges&#39;:
## 
##     shift
## 
## The following objects are masked from &#39;package:S4Vectors&#39;:
## 
##     first, second
## 
## The following objects are masked from &#39;package:lubridate&#39;:
## 
##     hour, isoweek, mday, minute, month, quarter, second, wday, week,
##     yday, year
## 
## The following object is masked from &#39;package:purrr&#39;:
## 
##     transpose
## 
## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     between, first, last
</code></pre></div>
<h2 id="metadata-input">Metadata input</h2>
<h3 id="identify-paths-for-metadata-and-project-data">Identify paths for metadata and project data</h3>
<p>Each user needs to write in their specific directory outputs prior to
the file name. The default working directory is this document so the
folder where this script is saved for the user. To change the workign
directory to the Rproject directory, select ‘Knit’ and ‘Knit Directory’
&gt; ‘Project Directory’.</p>
<div class="highlight"><pre><span></span><code><span class="c1">### User edits:</span>
<span class="c1">### 1. change paths of input and output as desired </span>

<span class="c1">## GMGI Fish database</span>
<span class="n">path_GMGIdb</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;C:/BoxDrive/Box/Science/Fisheries/Projects/eDNA/Metabarcoding Lab Resources/Reference Databases/GMGI_Vert_Ref.xlsx&quot;</span>

<span class="c1">## BLAST results</span>
<span class="n">path_blast_gmgi</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;example_input/BLASTResults_GMGI.txt&quot;</span>
<span class="n">path_blast_mito</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;example_input/BLASTResults_Mito.txt&quot;</span>
<span class="n">path_blast_ncbi_taxassigned</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;example_input/NCBI_taxassigned.txt&quot;</span>
<span class="n">path_blast_ncbi</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;example_input/BLASTResults_NCBI.txt&quot;</span>

<span class="c1">## ASV table results </span>
<span class="c1">## confirm that the ASV_table.len.tsv name is correct for user&#39;s project</span>
<span class="n">path_asv_table</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;example_input/ASV_table.len.tsv&quot;</span>
<span class="n">path_seq_file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;example_input/ASV_seqs.len.fasta&quot;</span>
<span class="n">path_output_summary</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;example_input/overall_summary.tsv&quot;</span>
</code></pre></div>
<p>Create output files</p>
<div class="highlight"><pre><span></span><code><span class="c1">### TAXONOMIC CHOICES </span>
<span class="c1"># output paths </span>
<span class="n">path_choice_required</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;example_output/Taxonomic_assignments/Choice_required_GMGI_multiplehits.xlsx&quot;</span>
<span class="n">path_choice_required_edited</span><span class="o">=</span><span class="s">&quot;example_output/Taxonomic_assignments/Choice_required_GMGI_multiplehits_edited.xlsx&quot;</span>

<span class="n">path_disagree_list</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;example_output/Taxonomic_assignments/SampleReport_taxonomic_ID.xlsx&quot;</span>
<span class="n">path_disagree_list_edited</span><span class="o">=</span><span class="s">&quot;example_output/Taxonomic_assignments/SampleReport_taxonomic_ID_edited.xlsx&quot;</span>

<span class="n">path_commonnames_add</span><span class="o">=</span><span class="s">&quot;example_output/Taxonomic_assignments/CommonNames_required.xlsx&quot;</span>
<span class="n">path_commonnames_add_edited</span><span class="o">=</span><span class="s">&quot;example_output/Taxonomic_assignments/CommonNames_required_edited.xlsx&quot;</span>

<span class="c1">#### RESULTS MATRICES </span>
<span class="n">results_rawreads_matrix</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;example_output/Results_1_rawreads_matrix.xlsx&quot;</span>
<span class="n">results_rawreads_long</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;example_output/Results_rawreads_long_format.xlsx&quot;</span>
<span class="n">results_relab_matrix</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;example_output/Results_2_relative_abundance_matrix.xlsx&quot;</span>
<span class="n">results_relab_long</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;example_output/Results_relative_abundance_long_format.xlsx&quot;</span>

<span class="n">reads_filtered_percent_out</span><span class="o">=</span><span class="s">&quot;example_output/Filtering_Percent_removed_reads.xlsx&quot;</span>
<span class="n">reads_filtered10_out</span><span class="o">=</span><span class="s">&quot;example_output/Filtering_10_removed_reads.xlsx&quot;</span>

<span class="n">Species_breakdown_sheet</span><span class="o">=</span><span class="s">&quot;example_output/Breakdown_Species_level.xlsx&quot;</span>
<span class="n">ASV_breakdown_sheet</span><span class="o">=</span><span class="s">&quot;example_output/Breakdown_ASV_level.xlsx&quot;</span>
</code></pre></div>
<h3 id="load-project-metadata">Load project metadata</h3>
<p>Metadata specific to each project. This contains information about each
sample (e.g., month, site, time, sample type, etc.). Confirm that sample
IDs match those used in the ASV_table.len.tsv file.</p>
<div class="highlight"><pre><span></span><code><span class="c1">### User edits:</span>
<span class="c1">### 1. change path of metadata file</span>

<span class="c1">## EXCEL</span>
<span class="c1"># meta &lt;- read_excel(&quot;example_input/metadata.xlsx&quot;)</span>

<span class="c1">## CSV </span>
<span class="n">meta</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.csv</span><span class="p">(</span><span class="s">&quot;example_input/SBNMS_metadata.csv&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## comment out if not needed</span>
<span class="w">  </span><span class="c1">## ampliseq processes _ not -</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">sampleID</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;-&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;_&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">sampleID</span><span class="p">))</span>
</code></pre></div>
<h3 id="load-database-metadata">Load database metadata</h3>
<p>No user edits in this section because paths have already been set above.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Load GMGI database information (common name, species name, etc.)</span>
<span class="n">gmgi_db</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read_xlsx</span><span class="p">(</span><span class="n">path_GMGIdb</span><span class="p">,</span><span class="w"> </span><span class="n">sheet</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">rename</span><span class="p">(</span><span class="n">sseqid</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Ref</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="c1">## removing &gt; from beginning of entires within Ref column</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">sseqid</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;&gt;&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">sseqid</span><span class="p">))</span>
</code></pre></div>
<h2 id="blast-data-input">BLAST data input</h2>
<p>No user edits unless user changed blastn parameters from fisheries team
default.</p>
<div class="highlight"><pre><span></span><code><span class="c1">## Setting column header names and classes</span>
<span class="n">blast_col_headers</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;ASV_ID&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;sseqid&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;pident&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;length&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;mismatch&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;gapopen&quot;</span><span class="p">,</span>
<span class="w">                                        </span><span class="s">&quot;qstart&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;qend&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;sstart&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;send&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;evalue&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;bitscore&quot;</span><span class="p">)</span>
<span class="n">blast_col_classes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="s">&quot;character&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">2</span><span class="p">),</span><span class="w"> </span><span class="nf">rep</span><span class="p">(</span><span class="s">&quot;numeric&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">10</span><span class="p">))</span>
</code></pre></div>
<h3 id="gmgi-database">GMGI database</h3>
<p>No user edits.</p>
<div class="highlight"><pre><span></span><code><span class="n">Blast_GMGI</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.table</span><span class="p">(</span><span class="n">path_blast_gmgi</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">col.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">blast_col_headers</span><span class="p">,</span><span class="w"> </span><span class="n">colClasses</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">blast_col_classes</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="c1">## blast changes spaces to hyphons so we need to change that back to match our metadata</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">sseqid</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;-&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot; &quot;</span><span class="p">,</span><span class="w"> </span><span class="n">sseqid</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## join with GMGI database information</span>
<span class="w">  </span><span class="nf">left_join</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">gmgi_db</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;sseqid&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## only keep the first human hit if there are multiple (this is already ordered by pident)</span>
<span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="p">(</span><span class="nf">str_detect</span><span class="p">(</span><span class="n">sseqid</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Human&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="nf">row_number</span><span class="p">()</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">1</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">ungroup</span><span class="p">()</span>

<span class="c1">## Check how many ASVs were identified with the GMGI Database</span>
<span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">Blast_GMGI</span><span class="o">$</span><span class="n">ASV_ID</span><span class="p">))</span><span class="w"> </span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 142
</code></pre></div>
<h3 id="mitofish-database">Mitofish database</h3>
<p>No user edits.</p>
<div class="highlight"><pre><span></span><code><span class="n">Blast_Mito</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.table</span><span class="p">(</span><span class="n">path_blast_mito</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">col.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">blast_col_headers</span><span class="p">,</span><span class="w"> </span><span class="n">colClasses</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">blast_col_classes</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="c1"># renaming sseqid to species name</span>
<span class="w">  </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">rename</span><span class="p">(</span><span class="n">Species_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">sseqid</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1"># replacing _ with spaces</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Species_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">gsub</span><span class="p">(</span><span class="s">&quot;_&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot; &quot;</span><span class="p">,</span><span class="w"> </span><span class="n">Species_name</span><span class="p">),</span>

<span class="w">         </span><span class="c1">## removing gb || sequence from species name </span>
<span class="w">         </span><span class="c1">### uncomment if we need this line again </span>
<span class="w">         </span><span class="c1"># Species_name = str_after_nth(Species_name, &quot;\\|&quot;, 2)</span>

<span class="w">         </span><span class="p">)</span>
</code></pre></div>
<h3 id="ncbi-database">NCBI database</h3>
<p>No user edits.</p>
<div class="highlight"><pre><span></span><code><span class="n">NCBI_taxassigned</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.delim2</span><span class="p">(</span><span class="n">path_blast_ncbi_taxassigned</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span><span class="w"> </span><span class="n">col.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;staxid&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Phylo&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="c1">## creating taxonomic assignment columns</span>
<span class="w">  </span><span class="nf">separate</span><span class="p">(</span><span class="n">Phylo</span><span class="p">,</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;Kingdom&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Phylum&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Class&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Order&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Family&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Genus&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Species_name&quot;</span><span class="p">),</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;;&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="c1">## creating species column based on Species_name</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">species</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">str_after_nth</span><span class="p">(</span><span class="n">Species_name</span><span class="p">,</span><span class="w"> </span><span class="s">&quot; &quot;</span><span class="p">,</span><span class="w"> </span><span class="m">1</span><span class="p">))</span>

<span class="n">Blast_NCBI</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.table</span><span class="p">(</span><span class="n">path_blast_ncbi</span><span class="p">,</span><span class="w"> </span><span class="n">header</span><span class="o">=</span><span class="bp">F</span><span class="p">,</span>
<span class="w">                           </span><span class="n">col.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;ASV_ID&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;sseqid&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;sscinames&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;staxid&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;pident&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;length&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;mismatch&quot;</span><span class="p">,</span>
<span class="w">                                         </span><span class="s">&quot;gapopen&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;qstart&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;qend&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;sstart&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;send&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;evalue&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;bitscore&quot;</span><span class="p">),</span>
<span class="w">                           </span><span class="n">colClasses</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="nf">rep</span><span class="p">(</span><span class="s">&quot;character&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">3</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;integer&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">rep</span><span class="p">(</span><span class="s">&quot;numeric&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">9</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">left_join</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">NCBI_taxassigned</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;staxid&quot;</span><span class="p">)</span>
</code></pre></div>
<h2 id="load-asv-sequences">Load ASV sequences</h2>
<p>No user edits.</p>
<div class="highlight"><pre><span></span><code><span class="c1">### Read in fasta file </span>
<span class="n">fasta</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">readDNAStringSet</span><span class="p">(</span><span class="n">path_seq_file</span><span class="p">)</span>

<span class="c1">## Creating db from the fasta information </span>
<span class="n">fasta_db</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.table</span><span class="p">(</span>
<span class="w">  </span><span class="n">ASV_ID</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">names</span><span class="p">(</span><span class="n">fasta</span><span class="p">),</span>
<span class="w">  </span><span class="n">sequence</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">fasta</span><span class="p">)</span>
<span class="p">)</span>
</code></pre></div>
<h2 id="load-dada2-asv-table">Load DADA2 ASV Table</h2>
<p>The column headers will be the Sample IDs and the first column is the
ASV ID. ASVs are given a “rank” based on sum of reads from that ASV
(pre-filtering). ‘Random’ indicates that if ASVs are tied, then the code
will randomly assign a rank for those tied. Because we don’t need an
exact rank here, ‘random’ will do for a tie-breaker.</p>
<p>No user edits.</p>
<div class="highlight"><pre><span></span><code><span class="n">ASV_table</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read_tsv</span><span class="p">(</span><span class="n">path_asv_table</span><span class="p">,</span><span class="w"> </span><span class="n">show_col_types</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="c1">## calculate the sum of all reads for each ASV</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">ASV_sum</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rowSums</span><span class="p">(</span><span class="nf">across</span><span class="p">(</span><span class="nf">where</span><span class="p">(</span><span class="n">is.numeric</span><span class="p">))))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>

<span class="w">  </span><span class="c1">## calculate a ranking based on those sum calculated above</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">ASV_rank</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">rank</span><span class="p">(</span><span class="o">-</span><span class="n">ASV_sum</span><span class="p">,</span><span class="w"> </span><span class="n">ties.method</span><span class="o">=</span><span class="s">&#39;random&#39;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## move the sum and rank columns to after ASV_ID and arrange by rank</span>
<span class="w">  </span><span class="nf">relocate</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">ASV_sum</span><span class="p">,</span><span class="n">ASV_rank</span><span class="p">),</span><span class="w"> </span><span class="n">.after</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ASV_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">arrange</span><span class="p">((</span><span class="n">ASV_rank</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## editing the rank value to be out of the max # </span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">ASV_rank</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">ASV_rank</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;/&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="n">ASV_rank</span><span class="p">,</span><span class="w"> </span><span class="n">na.rm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)))</span>

<span class="c1">## creating list of rankings</span>
<span class="n">ASV_rank_list</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ASV_table</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">,</span><span class="w"> </span><span class="n">ASV_sum</span><span class="p">,</span><span class="w"> </span><span class="n">ASV_rank</span><span class="p">)</span>
</code></pre></div>
<h2 id="taxonomic-assignment">Taxonomic Assignment</h2>
<p>Identifying where NCBI, Mito, and GMGI disagree on tax assignment. With
the hierarchial approach, ASVs that match to GMGI and several other
databases will only result in GMGI assignment. By reviewing this df, we
can be sure we aren’t missing an assignment in our GMGI curated
database.</p>
<p><strong>Sub-workflow:</strong><br />
1. Identify any ASVs that contain multiple hits within the GMGI
database.<br />
2. Identify entries that mismatch between GMGI, Mitofish, and NCBI
databases.<br />
3. Assign taxonomy based on hierarchical approach.<br />
4. Edit taxonomy annotations based on mismatch table.<br />
5. Adjusting common name for those entries that don’t have one (from
Mito or GMGI).</p>
<h3 id="identify-any-asvs-that-contain-multiple-hits-within-the-gmgi-database">Identify any ASVs that contain multiple hits within the GMGI database</h3>
<p>At this point, a fisheries team member needs to make choices about which
taxonomic assignment to accept.</p>
<h4 id="create-list-of-those-asvs-with-multiple-hits">Create list of those ASVs with multiple hits</h4>
<p>No user edits.</p>
<div class="highlight"><pre><span></span><code><span class="n">multiple_hit_choice</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Blast_GMGI</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">group_by</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="c1">## take top percent identity hit, count the number of top hits, and filter to those with more than 1 top hit </span>
<span class="w">  </span><span class="nf">slice_max</span><span class="p">(</span><span class="n">pident</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="o">=</span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">count</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">n</span><span class="o">&gt;</span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">rename</span><span class="p">(</span><span class="n">No_tied_hits</span><span class="o">=</span><span class="n">n</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## adding BLAST_GMGI information with these ASVs and ASV rank and sum</span>
<span class="w">  </span><span class="nf">left_join</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">Blast_GMGI</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ASV_ID&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">left_join</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">ASV_rank_list</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ASV_ID&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## adding sequencing information </span>
<span class="w">  </span><span class="nf">left_join</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">fasta_db</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ASV_ID&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## moving database percent ID to be next to Blast percent ID</span>
<span class="w">  </span><span class="nf">relocate</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">db_percent_ID</span><span class="p">,</span><span class="w"> </span><span class="n">ASV_sum</span><span class="p">,</span><span class="w"> </span><span class="n">ASV_rank</span><span class="p">),</span><span class="w"> </span><span class="n">.after</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pident</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## adding choice column for next steps </span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Choice</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">)</span>

<span class="c1">## export this data frame as excel sheet </span>
<span class="n">multiple_hit_choice</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">write_xlsx</span><span class="p">(</span><span class="n">path_choice_required</span><span class="p">)</span>
</code></pre></div>
<p>Based on the output above, user needs to make some choices. In the excel
spreadsheet, user needs to mark ‘x’ on the choices desired while leaving
the other entries blank.</p>
<h4 id="choosing-one-of-several-hits">Choosing one of several hits.</h4>
<p>Load choice edited dataset. No user edits.</p>
<div class="highlight"><pre><span></span><code><span class="n">multiple_hit_choice_edited</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read_xlsx</span><span class="p">(</span><span class="n">path_choice_required_edited</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="c1">## selecting the choices made</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">Choice</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="c1">## selecting only columns needed </span>
<span class="w">  </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">,</span><span class="w"> </span><span class="n">sseqid</span><span class="p">,</span><span class="w"> </span><span class="n">Choice</span><span class="p">)</span>
</code></pre></div>
<p>A for loop will filter Blast_GMGI df based on these choices. No user
edits.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># Create a new edited df</span>
<span class="n">Blast_GMGI_edited</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Blast_GMGI</span><span class="w"> </span>

<span class="c1"># Loop through each row of the dataframe</span>
<span class="nf">for </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">multiple_hit_choice_edited</span><span class="o">$</span><span class="n">ASV_ID</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="c1"># Extract the current row (will do this for each ASV_ID in the choice df)</span>
<span class="w">  </span><span class="n">current_row</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">multiple_hit_choice_edited</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">subset</span><span class="p">(</span><span class="n">ASV_ID</span><span class="o">==</span><span class="n">i</span><span class="p">)</span>

<span class="w">  </span><span class="c1"># Apply filter based on the current row&#39;s condition</span>
<span class="w">  </span><span class="n">Blast_GMGI_edited</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Blast_GMGI_edited</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">filter</span><span class="p">(</span><span class="nf">case_when</span><span class="p">(</span><span class="n">ASV_ID</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">current_row</span><span class="o">$</span><span class="n">ASV_ID</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">sseqid</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">current_row</span><span class="o">$</span><span class="n">sseqid</span><span class="p">,</span>
<span class="w">           </span><span class="kc">TRUE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span>
<span class="p">}</span>
</code></pre></div>
<h4 id="confirming-that-all-entries-have-been-dealth-with">Confirming that all entries have been dealth with</h4>
<p>No user edits.</p>
<div class="highlight"><pre><span></span><code><span class="c1">### Check the below output to confirm the filtering steps above worked (if it worked, it won&#39;t be in output)</span>
<span class="n">Blast_GMGI_edited</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">group_by</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">slice_max</span><span class="p">(</span><span class="n">pident</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="o">=</span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">count</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">n</span><span class="o">&gt;</span><span class="m">1</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## # A tibble: 0 × 2
## # Groups:   ASV_ID [0]
## # ℹ 2 variables: ASV_ID &lt;chr&gt;, n &lt;int&gt;
</code></pre></div>
<h3 id="identify-entries-that-mismatch-between-gmgi-mitofish-and-ncbi-databases">Identify entries that mismatch between GMGI, Mitofish, and NCBI databases</h3>
<p>Creating a df called “Disagree”. Review the output before moving onto
the next section.</p>
<p>No user edits.</p>
<div class="highlight"><pre><span></span><code><span class="c1">## Create GMGI species input</span>
<span class="n">disagree_input_GMGI</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Blast_GMGI_edited</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">,</span><span class="w"> </span><span class="n">GMGI_pident</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pident</span><span class="p">,</span><span class="w"> </span><span class="n">GMGI_db_ID</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">db_percent_ID</span><span class="p">,</span><span class="w"> </span><span class="n">GMGI_Species</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Species_name</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">slice_max</span><span class="p">(</span><span class="n">GMGI_pident</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="n">with_ties</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">ungroup</span><span class="p">()</span>

<span class="c1">## Create Mitofish input</span>
<span class="n">disagree_input_Mito</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Blast_Mito</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">,</span><span class="w"> </span><span class="n">Mito_Species</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Species_name</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">distinct</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">group_by</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Mito_Species</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">Mito_Species</span><span class="p">,</span><span class="w"> </span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;;&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">distinct</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">ungroup</span><span class="p">()</span>

<span class="c1">## Create NCBI input</span>
<span class="n">disagree_input_NCBI</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">Blast_NCBI</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">,</span><span class="w"> </span><span class="n">NCBI_Species</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Species_name</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">distinct</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">NCBI_Species</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">NCBI_Species</span><span class="p">,</span><span class="w"> </span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;;&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">distinct</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">ungroup</span><span class="p">()</span>

<span class="c1">## Combine all three dfs into disagree_df with ASV rank information</span>
<span class="n">disagree_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">disagree_input_GMGI</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">full_join</span><span class="p">(</span><span class="n">disagree_input_Mito</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ASV_ID&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">full_join</span><span class="p">(</span><span class="n">disagree_input_NCBI</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ASV_ID&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">full_join</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">ASV_rank_list</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ASV_ID&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">left_join</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">fasta_db</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ASV_ID&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Choice</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">)</span>

<span class="c1">## Filtering the disagree_df to only output the entries that disagree with multiple entries</span>
<span class="n">filtered_disagree_df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">disagree_df</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## grouping by ASV (row)</span>
<span class="w">  </span><span class="nf">rowwise</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## Filtering out rows that are unassigned across all 3 db </span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="w"> </span><span class="p">(</span><span class="nf">is.na</span><span class="p">(</span><span class="n">GMGI_Species</span><span class="p">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nf">is.na</span><span class="p">(</span><span class="n">Mito_Species</span><span class="p">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nf">is.na</span><span class="p">(</span><span class="n">NCBI_Species</span><span class="p">)</span><span class="w"> </span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## Filter out rows that are GMGI assigned and empty from Mito and NCBI</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="o">!</span><span class="w"> </span><span class="p">(</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">GMGI_Species</span><span class="p">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nf">is.na</span><span class="p">(</span><span class="n">Mito_Species</span><span class="p">)</span><span class="w"> </span><span class="o">&amp;&amp;</span><span class="w"> </span><span class="nf">is.na</span><span class="p">(</span><span class="n">NCBI_Species</span><span class="p">)</span><span class="w"> </span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## Filtering out rows that have the same information across databases (ONLY in rows with a GMGI entry)</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="nf">ifelse</span><span class="p">(</span><span class="o">!</span><span class="nf">is.na</span><span class="p">(</span><span class="n">GMGI_Species</span><span class="p">),</span><span class="w"> </span>
<span class="w">          </span><span class="nf">n_distinct</span><span class="p">(</span><span class="nf">c_across</span><span class="p">(</span><span class="n">GMGI_Species</span><span class="o">:</span><span class="n">NCBI_Species</span><span class="p">),</span><span class="w"> </span><span class="n">na.rm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span>

<span class="c1">## export this data frame as excel sheet </span>
<span class="n">filtered_disagree_df</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">write_xlsx</span><span class="p">(</span><span class="n">path_disagree_list</span><span class="p">)</span>
</code></pre></div>
<h3 id="assign-taxonomy-based-on-hierarchical-approach">Assign taxonomy based on hierarchical approach</h3>
<p>Taxonomic identification is taken from GMGI 100%, then GMGI \&lt;100%, then
Mitofish 100%, and finally NCBI 100%.</p>
<p>No user edits.</p>
<div class="highlight"><pre><span></span><code><span class="n">ASV_table_taxID</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ASV_table</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>

<span class="w">  </span><span class="c1">## 1. Top hit from GMGI&#39;s database</span>
<span class="w">  </span><span class="nf">left_join</span><span class="p">(</span><span class="n">Blast_GMGI_edited</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">              </span><span class="nf">slice_max</span><span class="p">(</span><span class="n">pident</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">                            </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">,</span><span class="w"> </span><span class="n">Species_name</span><span class="p">),</span>
<span class="w">            </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">join_by</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## 2. Mitofish database</span>
<span class="w">  </span><span class="c1">### join df, select ASV_ID and Species_name columns, rename Species_name to Mito, call only distinct rows</span>
<span class="w">  </span><span class="nf">left_join</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">Blast_Mito</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">,</span><span class="w"> </span><span class="n">Mito</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Species_name</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">distinct</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">              </span><span class="c1">### group by ASV_ID, and collapse all species names separated by ;, then take only distinct rows</span>
<span class="w">              </span><span class="nf">group_by</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Mito</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">Mito</span><span class="p">,</span><span class="w"> </span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;;&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">distinct</span><span class="p">(),</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ASV_ID&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">### if GMGI annotation is NA, then replace with Mitofish </span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">Species_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="nf">is.na</span><span class="p">(</span><span class="n">Species_name</span><span class="p">),</span><span class="w"> </span><span class="n">Mito</span><span class="p">,</span><span class="w"> </span><span class="n">Species_name</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## 3. NCBI database; same functions as above</span>
<span class="w">  </span><span class="nf">left_join</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">Blast_NCBI</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">,</span><span class="w"> </span><span class="n">NCBI</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Species_name</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">distinct</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">              </span><span class="nf">group_by</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">mutate</span><span class="p">(</span><span class="n">NCBI</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">paste0</span><span class="p">(</span><span class="n">NCBI</span><span class="p">,</span><span class="w"> </span><span class="n">collapse</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;;&quot;</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">distinct</span><span class="p">(),</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;ASV_ID&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">Species_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="nf">is.na</span><span class="p">(</span><span class="n">Species_name</span><span class="p">),</span><span class="w"> </span><span class="n">NCBI</span><span class="p">,</span><span class="w"> </span><span class="n">Species_name</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## 4. if Species name is STILL not filled, call it &quot;Unassigned&quot;</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">Species_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="nf">is.na</span><span class="p">(</span><span class="n">Species_name</span><span class="p">),</span><span class="w"> </span><span class="s">&quot;Unassigned&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">Species_name</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">  </span>

<span class="w">  </span><span class="c1">## removing Mito spp and NCBI spp</span>
<span class="w">  </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">Mito</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">NCBI</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## move species name to be after ASV_ID</span>
<span class="w">  </span><span class="nf">relocate</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">Species_name</span><span class="p">),</span><span class="w"> </span><span class="n">.after</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ASV_ID</span><span class="p">)</span>
</code></pre></div>
<h3 id="edit-taxonomy-annotations-based-on-mismatch-table">Edit taxonomy annotations based on mismatch table</h3>
<p>Override any annotations with edited taxonomic identification table.<br />
No user edits.</p>
<div class="highlight"><pre><span></span><code><span class="c1">## read in edited df </span>
<span class="n">taxonomic_choice</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read_xlsx</span><span class="p">(</span><span class="n">path_disagree_list_edited</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="c1">## selecting only columns needed </span>
<span class="w">  </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">,</span><span class="w"> </span><span class="n">Choice</span><span class="p">)</span><span class="w">  </span>

<span class="c1"># Create a new edited df</span>
<span class="n">ASV_table_taxID_edited</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ASV_table_taxID</span><span class="w"> </span>

<span class="c1"># Loop through each row of the dataframe</span>
<span class="nf">for </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">taxonomic_choice</span><span class="o">$</span><span class="n">ASV_ID</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="c1"># Extract the current row (will do this for each ASV_ID in the choice df)</span>
<span class="w">  </span><span class="n">current_row</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">taxonomic_choice</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">subset</span><span class="p">(</span><span class="n">ASV_ID</span><span class="o">==</span><span class="n">i</span><span class="p">)</span>

<span class="w">  </span><span class="c1"># Apply filter based on the current row&#39;s condition</span>
<span class="w">  </span><span class="n">ASV_table_taxID_edited</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ASV_table_taxID_edited</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Species_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">case_when</span><span class="p">(</span>
<span class="w">          </span><span class="n">ASV_ID</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">current_row</span><span class="o">$</span><span class="n">ASV_ID</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">current_row</span><span class="o">$</span><span class="n">Choice</span><span class="p">,</span>
<span class="w">           </span><span class="kc">TRUE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">Species_name</span><span class="p">))</span>
<span class="p">}</span>
</code></pre></div>
<h4 id="confirm-all-entries-are-dealt-with">Confirm all entries are dealt with</h4>
<p>No user edits.</p>
<div class="highlight"><pre><span></span><code><span class="c1">## Output will be blank</span>
<span class="n">ASV_table_taxID_edited</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">Species_name</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">distinct</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="nf">grepl</span><span class="p">(</span><span class="s">&quot;;&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">Species_name</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">arrange</span><span class="p">(</span><span class="n">Species_name</span><span class="p">)</span><span class="w"> </span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## # A tibble: 0 × 1
## # ℹ 1 variable: Species_name &lt;chr&gt;
</code></pre></div>
<h3 id="adjusting-common-name-for-those-entries-that-dont-have-one-from-mito-or-ncbi">Adjusting common name for those entries that don’t have one (from Mito or NCBI)</h3>
<p>No user edits.</p>
<div class="highlight"><pre><span></span><code><span class="c1">### add common name column to df</span>
<span class="n">ASV_table_taxID_edited</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ASV_table_taxID_edited</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">left_join</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">gmgi_db</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">Species_name</span><span class="p">,</span><span class="w"> </span><span class="n">Common_name</span><span class="p">,</span><span class="w"> </span><span class="n">Category</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">distinct</span><span class="p">(),</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Species_name&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">relocate</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">Common_name</span><span class="p">,</span><span class="w"> </span><span class="n">Category</span><span class="p">),</span><span class="w"> </span><span class="n">.after</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Species_name</span><span class="p">)</span>

<span class="c1">### print entries with no common name</span>
<span class="n">ASV_table_taxID_edited</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">Species_name</span><span class="p">,</span><span class="w"> </span><span class="n">Common_name</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="nf">is.na</span><span class="p">(</span><span class="n">Common_name</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">distinct</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Category</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">Kingdom</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">Phylum</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">Class</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">Order</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">Family</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">Genus</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">species</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">NA</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span>
<span class="w">    </span><span class="nf">across</span><span class="p">(</span><span class="nf">everything</span><span class="p">(),</span><span class="w"> </span><span class="o">~</span><span class="nf">case_when</span><span class="p">(</span>
<span class="w">      </span><span class="n">Species_name</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s">&quot;Unassigned&quot;</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s">&quot;Unassigned&quot;</span><span class="p">,</span>
<span class="w">      </span><span class="kc">TRUE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">.x</span>
<span class="w">    </span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">write_xlsx</span><span class="p">(</span><span class="n">path_commonnames_add</span><span class="p">)</span>
</code></pre></div>
<p>Editing common names and category when needed.</p>
<div class="highlight"><pre><span></span><code><span class="c1">## read in edited df </span>
<span class="n">commonNames_annotated</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read_xlsx</span><span class="p">(</span><span class="n">path_commonnames_add_edited</span><span class="p">)</span>

<span class="c1"># Create a new edited df</span>
<span class="n">ASV_table_taxID_annotated</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ASV_table_taxID_edited</span><span class="w"> </span>

<span class="c1"># Loop through each row of the dataframe</span>
<span class="nf">for </span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">commonNames_annotated</span><span class="o">$</span><span class="n">Species_name</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="c1"># Extract the current row (will do this for each ASV_ID in the choice df)</span>
<span class="w">  </span><span class="n">current_row</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">commonNames_annotated</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">subset</span><span class="p">(</span><span class="n">Species_name</span><span class="o">==</span><span class="n">i</span><span class="p">)</span>

<span class="w">  </span><span class="c1"># Apply filter based on the current row&#39;s condition</span>
<span class="w">  </span><span class="n">ASV_table_taxID_annotated</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ASV_table_taxID_annotated</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Common_name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">case_when</span><span class="p">(</span>
<span class="w">          </span><span class="n">Species_name</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">current_row</span><span class="o">$</span><span class="n">Species_name</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">current_row</span><span class="o">$</span><span class="n">Common_name</span><span class="p">,</span>
<span class="w">           </span><span class="kc">TRUE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">Common_name</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">    </span><span class="nf">mutate</span><span class="p">(</span><span class="n">Category</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">case_when</span><span class="p">(</span>
<span class="w">          </span><span class="n">Species_name</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="n">current_row</span><span class="o">$</span><span class="n">Species_name</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">current_row</span><span class="o">$</span><span class="n">Category</span><span class="p">,</span>
<span class="w">           </span><span class="kc">TRUE</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">Category</span><span class="p">))</span><span class="w">  </span>
<span class="p">}</span>

<span class="c1">## printing list of species name without common names </span>
<span class="c1">## after additions to mutate function above, this output should be zero </span>
<span class="n">ASV_table_taxID_annotated</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">Species_name</span><span class="p">,</span><span class="w"> </span><span class="n">Common_name</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="nf">is.na</span><span class="p">(</span><span class="n">Common_name</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">distinct</span><span class="p">()</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## # A tibble: 0 × 2
## # ℹ 2 variables: Species_name &lt;chr&gt;, Common_name &lt;chr&gt;
</code></pre></div>
<h2 id="filtering">Filtering</h2>
<h3 id="filter-out-reads-that-are-less-than-10-in-each-sample-per-asv">Filter out reads that are less than 10 in each sample per ASV</h3>
<div class="highlight"><pre><span></span><code><span class="n">ASV_table_taxID_filtered_step1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ASV_table_taxID_annotated</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="c1">## telling the df we are doing the following function by rows (ASVs)</span>
<span class="w">  </span><span class="nf">rowwise</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## Filter out reads that are less than 10 in each sample per ASV</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="nf">across</span><span class="p">(</span><span class="n">.cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="m">7</span><span class="o">:</span><span class="nf">ncol</span><span class="p">(</span><span class="n">.</span><span class="p">)),</span>
<span class="w">                </span><span class="n">.fns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">.x</span><span class="o">&lt;</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">.x</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">ungroup</span><span class="p">()</span>

<span class="nf">nrow</span><span class="p">(</span><span class="n">ASV_table_taxID_filtered_step1</span><span class="p">)</span><span class="w">  </span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 158
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="c1">## Remove rows with zero sums across the table</span>
<span class="n">ASV_table_taxID_filtered_step2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ASV_table_taxID_filtered_step1</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="c1"># Keep only rows where the sum across columns 7:ncol(.) is greater than 0</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="nf">rowSums</span><span class="p">(</span><span class="nf">across</span><span class="p">(</span><span class="m">7</span><span class="o">:</span><span class="nf">ncol</span><span class="p">(</span><span class="n">.</span><span class="p">)))</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">)</span>

<span class="nf">nrow</span><span class="p">(</span><span class="n">ASV_table_taxID_filtered_step2</span><span class="p">)</span><span class="w"> </span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 153
</code></pre></div>
<p>Create output of reads that were filtered out</p>
<div class="highlight"><pre><span></span><code><span class="n">ASV_table_taxID_annotated</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="c1">## telling the df we are doing the following function by rows (ASVs)</span>
<span class="w">  </span><span class="nf">rowwise</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## Filter out reads that are less than 10 in each sample per ASV</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="nf">across</span><span class="p">(</span><span class="n">.cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="m">7</span><span class="o">:</span><span class="nf">ncol</span><span class="p">(</span><span class="n">.</span><span class="p">)),</span>
<span class="w">                </span><span class="n">.fns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">(</span><span class="n">.x</span><span class="o">&gt;</span><span class="m">10</span><span class="p">,</span><span class="w"> </span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">.x</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">ungroup</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="nf">write_xlsx</span><span class="p">(</span><span class="n">reads_filtered10_out</span><span class="p">)</span>
</code></pre></div>
<h3 id="filter-out-reads-that-are-less-than-a-certain-of-asv-row-total-per-sample">Filter out reads that are less than a certain % of ASV (row) total per sample</h3>
<p>Values are based on the MiSeq used because the flow cell elicits
different adapter hopping tendencies:<br />
- Legacy MiSeq = 0.001<br />
- New MiSeq = 0.0025</p>
<div class="highlight"><pre><span></span><code><span class="n">ASV_table_taxID_filtered</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ASV_table_taxID_filtered_step2</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="c1">## telling the df we are doing the following function by rows (ASVs)</span>
<span class="w">  </span><span class="nf">rowwise</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="nf">across</span><span class="p">(</span><span class="n">.cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">7</span><span class="o">:</span><span class="nf">ncol</span><span class="p">(</span><span class="n">.</span><span class="p">),</span>
<span class="w">                </span><span class="n">.fns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">((</span><span class="n">.x</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">.x</span><span class="p">,</span><span class="w"> </span><span class="n">na.rm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">0.001</span><span class="p">,</span><span class="w"> </span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">.x</span><span class="p">)))</span>
</code></pre></div>
<p>Create output of reads that were filtered out</p>
<div class="highlight"><pre><span></span><code><span class="n">ASV_table_taxID_filtered_step1</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="c1">## telling the df we are doing the following function by rows (ASVs)</span>
<span class="w">  </span><span class="nf">rowwise</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="nf">across</span><span class="p">(</span><span class="n">.cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">7</span><span class="o">:</span><span class="nf">ncol</span><span class="p">(</span><span class="n">.</span><span class="p">),</span>
<span class="w">                </span><span class="n">.fns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="nf">ifelse</span><span class="p">((</span><span class="n">.x</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">.x</span><span class="p">,</span><span class="w"> </span><span class="n">na.rm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0.001</span><span class="p">,</span><span class="w"> </span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">.x</span><span class="p">)))</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="nf">write_xlsx</span><span class="p">(</span><span class="n">reads_filtered_percent_out</span><span class="p">)</span>
</code></pre></div>
<h3 id="filter-out-samples-with-less-than-x-number-of-reads">Filter out samples with less than X number of reads</h3>
<p>Remove samples that don’t meet minimum read count thresholds. GMGI
Fisheries team minimum is 1,000 reads.</p>
<div class="highlight"><pre><span></span><code><span class="c1">### View final number of reads for each sample</span>
<span class="n">colsums</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.data.frame</span><span class="p">(</span><span class="nf">colSums</span><span class="p">(</span><span class="n">ASV_table_taxID_filtered</span><span class="p">[,</span><span class="w"> </span><span class="m">7</span><span class="o">:</span><span class="nf">ncol</span><span class="p">(</span><span class="n">ASV_table_taxID_filtered</span><span class="p">)],</span><span class="w"> </span><span class="n">na.rm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">rename</span><span class="p">(</span><span class="n">sum</span><span class="o">=</span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">arrange</span><span class="p">(</span><span class="n">sum</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">rownames_to_column</span><span class="p">(</span><span class="n">var</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;sampleID&quot;</span><span class="p">)</span>

<span class="nf">range</span><span class="p">(</span><span class="n">colsums</span><span class="o">$</span><span class="n">sum</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1]      0 244276
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="c1">### Identify samples with &lt;1000 reads</span>
<span class="n">low_read_samples</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">colsums</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">filter</span><span class="p">(</span><span class="n">sum</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">1000</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">pull</span><span class="p">(</span><span class="n">sampleID</span><span class="p">)</span>

<span class="c1">### Remove those samples from dataset</span>
<span class="n">ASV_table_taxID_filtered</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ASV_table_taxID_filtered</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="nf">all_of</span><span class="p">(</span><span class="n">low_read_samples</span><span class="p">))</span>

<span class="c1">### Verify</span>
<span class="nf">range</span><span class="p">(</span><span class="nf">colSums</span><span class="p">(</span><span class="n">ASV_table_taxID_filtered</span><span class="p">[,</span><span class="w"> </span><span class="m">7</span><span class="o">:</span><span class="nf">ncol</span><span class="p">(</span><span class="n">ASV_table_taxID_filtered</span><span class="p">)],</span><span class="w"> </span><span class="n">na.rm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">))</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1]   9364 244276
</code></pre></div>
<h2 id="create-asv-breakdown-table">Create ASV breakdown table</h2>
<p>No user edits.</p>
<div class="highlight"><pre><span></span><code><span class="c1">## Export ASV break-down for 03-data_quality.Rmds</span>
<span class="n">ASV_table_taxID_filtered</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="n">ASV_ID</span><span class="p">,</span><span class="w"> </span><span class="n">Species_name</span><span class="p">,</span><span class="w"> </span><span class="n">Common_name</span><span class="p">,</span><span class="w"> </span><span class="n">Category</span><span class="p">,</span><span class="w"> </span><span class="n">ASV_sum</span><span class="p">,</span><span class="w"> </span><span class="n">ASV_rank</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">write_xlsx</span><span class="p">(</span><span class="n">ASV_breakdown_sheet</span><span class="p">)</span>

<span class="c1">## Confirm number of samples is as expected (outputs should be the same)</span>
<span class="nf">ncol</span><span class="p">(</span><span class="n">ASV_table_taxID_annotated</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">ASV_ID</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">Species_name</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">Common_name</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">Category</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">ASV_sum</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">ASV_rank</span><span class="p">))</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 178
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="nf">ncol</span><span class="p">(</span><span class="n">ASV_table_taxID_filtered</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">ASV_ID</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">Species_name</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">Common_name</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">Category</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">ASV_sum</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">ASV_rank</span><span class="p">))</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 142
</code></pre></div>
<h2 id="collapsing-read-counts-by-species-name">Collapsing read counts by species name</h2>
<p>No user edits.</p>
<div class="highlight"><pre><span></span><code><span class="n">ASV_table_taxID_collapsed</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ASV_table_taxID_filtered</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="c1"># removing original ASV_ID to collapse</span>
<span class="w">  </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">ASV_ID</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">ASV_sum</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">ASV_rank</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">  </span>

<span class="w">  </span><span class="c1">## group by Species_name and sample</span>
<span class="w">  </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">group_by</span><span class="p">(</span><span class="n">Species_name</span><span class="p">,</span><span class="w"> </span><span class="n">Common_name</span><span class="p">,</span><span class="w"> </span><span class="n">Category</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## sum down column by species name and sample to collapse</span>
<span class="w">  </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">summarise</span><span class="p">(</span><span class="nf">across</span><span class="p">(</span><span class="n">.cols</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="nf">last_col</span><span class="p">()),</span><span class="w">            </span>
<span class="w">                          </span><span class="n">.fns</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">na.rm</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)),</span><span class="w"> </span>
<span class="w">                     </span><span class="n">.groups</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;drop&#39;</span><span class="p">)</span>

<span class="c1"># Print number of samples = 187</span>
<span class="nf">ncol</span><span class="p">(</span><span class="n">ASV_table_taxID_collapsed</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">Species_name</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">Common_name</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">Category</span><span class="p">))</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 142
</code></pre></div>
<h2 id="creating-results-output">Creating results output</h2>
<p>Raw reads results output.<br />
No user edits.</p>
<div class="highlight"><pre><span></span><code><span class="c1">### Number of total assignments -1 for unassigned</span>
<span class="nf">length</span><span class="p">(</span><span class="nf">unique</span><span class="p">(</span><span class="n">ASV_table_taxID_collapsed</span><span class="o">$</span><span class="n">Species_name</span><span class="p">))</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="m">1</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 42
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="c1">## Raw reads matrix (wide format)</span>
<span class="n">ASV_table_taxID_collapsed</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">write_xlsx</span><span class="p">(</span><span class="n">results_rawreads_matrix</span><span class="p">)</span>

<span class="c1">## Raw reads long format and filtering out entries with zero reads</span>
<span class="n">ASV_table_taxID_collapsed</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">gather</span><span class="p">(</span><span class="s">&quot;sampleID&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;reads&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">4</span><span class="o">:</span><span class="nf">last_col</span><span class="p">()))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">filter</span><span class="p">(</span><span class="n">reads</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">0</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">left_join</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">meta</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;sampleID&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">write_xlsx</span><span class="p">(</span><span class="n">results_rawreads_long</span><span class="p">)</span>

<span class="c1">## exporting species summary</span>
<span class="n">ASV_table_taxID_collapsed</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">gather</span><span class="p">(</span><span class="s">&quot;sampleID&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;reads&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">4</span><span class="o">:</span><span class="nf">last_col</span><span class="p">()))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">Species_name</span><span class="p">,</span><span class="w"> </span><span class="n">Common_name</span><span class="p">,</span><span class="w"> </span><span class="n">Category</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="n">dplyr</span><span class="o">::</span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">sampleID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">summarise</span><span class="p">(</span><span class="n">sum_reads</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">reads</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">write_xlsx</span><span class="p">(</span><span class="n">Species_breakdown_sheet</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## `summarise()` has grouped output by &#39;Species_name&#39;, &#39;Common_name&#39;. You can
## override using the `.groups` argument.
</code></pre></div>
<p>Relative Abundance No user edits.</p>
<div class="highlight"><pre><span></span><code><span class="c1">### Calculating relative abundance</span>
<span class="n">df_relab</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ASV_table_taxID_collapsed</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span>
<span class="w">  </span><span class="nf">gather</span><span class="p">(</span><span class="s">&quot;sampleID&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;reads&quot;</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="o">:</span><span class="nf">last_col</span><span class="p">())</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">sampleID</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">### total</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">sample_total</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">reads</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">group_by</span><span class="p">(</span><span class="n">sampleID</span><span class="p">,</span><span class="w"> </span><span class="n">Species_name</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>

<span class="w">  </span><span class="c1">## relab</span>
<span class="w">  </span><span class="nf">mutate</span><span class="p">(</span><span class="n">relab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">reads</span><span class="o">/</span><span class="n">sample_total</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">ungroup</span><span class="p">()</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">select</span><span class="p">(</span><span class="o">-</span><span class="n">reads</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">sample_total</span><span class="p">)</span><span class="w"> </span>

<span class="n">df_relab</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">left_join</span><span class="p">(</span><span class="n">.</span><span class="p">,</span><span class="w"> </span><span class="n">meta</span><span class="p">,</span><span class="w"> </span><span class="n">by</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;sampleID&quot;</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span>
<span class="w">  </span><span class="nf">write_xlsx</span><span class="p">(</span><span class="n">results_relab_long</span><span class="p">)</span>

<span class="n">df_relab</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">spread</span><span class="p">(</span><span class="n">sampleID</span><span class="p">,</span><span class="w"> </span><span class="n">relab</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w"> </span><span class="nf">write_xlsx</span><span class="p">(</span><span class="n">results_relab_matrix</span><span class="p">)</span>
</code></pre></div>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["content.action.edit", "content.code.copy"], "search": "../../assets/javascripts/workers/search.6ce7567c.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.d6f25eb3.min.js"></script>
      
    
  </body>
</html>