## Collection of Roberts Lab protocols.

Brief description of repo files are below 

- **readme.md** - 

- **Homogenization of tissues:** - NEED TO CREATE PROTOCOL 

## DNA

- **[ethanol_precipitation_DNA.md](https://github.com/RobertsLab/resources/blob/master/protocols/ethanol_precipitation_DNA.md)** - Ethanol precipitation of DNA. Recommended for X and Y tissues.  Not recommended for Z. 

- **[RADseq.md](https://github.com/RobertsLab/resources/blob/master/protocols/RADseq.md)** - Preparing tissue for DNA sequencing using RADseq, which is commonly used for genetics experiments, such as identification of genetic variants (SNPs), phylogenetics, population structure, linkage and quantitative trait locus mapping, genome wide association studies, etc. COULD BE IMPROVED / UPDATED


## DNA methylation 

- **[medip.md](https://github.com/RobertsLab/resources/blob/master/protocols/medip.md)** - Protocol for methylated DNA immunoprecipitated.

- **[BSseq-Nov-2015.md](https://github.com/RobertsLab/resources/blob/master/protocols/BSseq-Nov-2015.md)** -

- **[BSseq.md](https://github.com/RobertsLab/resources/blob/master/protocols/BSseq.md)** -

## RNA

- **[rna_isolation_rnazol_rt.md](https://github.com/RobertsLab/resources/blob/master/protocols/rna_isolation_rnazol_rt.md)** - Total RNA isolation using RNAzol RT. Reliably isolates Total RNA from frozen tissues of relatively large size that have been properly preserved (by flash-freezing or using RNAlater), and tissues that can be homogenized thoroughly prior to processing with RNAzol. Not recommended for hemolymph, or very small amounts of tissue.  Protocol can also be used to isolate mRNA and micro RNA. 

- **[dnase_rna.md](https://github.com/RobertsLab/resources/blob/master/protocols/dnase_rna.md)** - DNase procedure using the Turbo DNA-free Kit.  This is commonly used to remove DNA contamination from isolated RNA, prior to generating cDNA. 

- **[reverse_transcription.md](https://github.com/RobertsLab/resources/blob/master/protocols/reverse_transcription.md)** - Reverse transcription using Promega M-MMLV Reverse Transcriptase. Used when 

- **[qpcr.md](https://github.com/RobertsLab/resources/blob/master/protocols/qpcr.md)** - Quantitative PCR protocol using SsoFast EvaGreen master mix.

- **[RNA_Pico_chip_bioanalyzer_protocol.md](https://github.com/RobertsLab/resources/blob/master/protocols/RNA_Pico_chip_bioanalyzer_protocol.md)** - SOP for the quantification of RNA quality using the RNA 6000 Pico Chip and Agilent Bioanalyzer

## Protein

- **[MassSpec_Tutorial.md](https://github.com/RobertsLab/resources/blob/master/protocols/MassSpec_Tutorial.md)** - Explanation of the different types of mass spectrometry (MS, or "mass spec") experiments one could use to measure protein abundances via "proteomics" 

- **[ProteinprepforMSMS.md](https://github.com/RobertsLab/resources/blob/master/protocols/ProteinprepforMSMS.md)** - Bench protocol for taking tissues that have been flash frozen, and extracting protein for proteomics experiments. 

- **[DIA-data-Analyses.md](https://github.com/RobertsLab/resources/blob/master/protocols/DIA-data-Analyses.md)** - Steps to analyze data derived from Data Independent Acquisition (DIA) runs on a mass spectrometer. Included are steps from taking raw data to exporting a .csv table of protein abundances for downstream statistical analysis (e.g. in RStudio). 

- **[SRM-data-Analyses.md](https://github.com/RobertsLab/resources/blob/master/protocols/SRM-data-Analyses.md)** -  Steps to analyze data derived from Selected Reaction Monitoring (SRM) runs on a mass spectrometer. Included are steps from taking raw data to exporting a .csv table of protein abundances for downstream statistical analysis (e.g. in RStudio).

## Enzymes 

- **[ATPase.md](https://github.com/RobertsLab/resources/blob/master/protocols/ATPase.md)** - SOP for the collection, extraction, and quantification of ATPase activity from oyster tissue samples.   

## Miscellaneous 

- **[setting_larvae.md](https://github.com/RobertsLab/resources/blob/master/protocols/setting_larvae.md)** - How to induce Pacific oyster larvae to metamorphose 

- **[Titrator.SOP.md](https://github.com/RobertsLab/titrator/blob/43cda74c50feab00a3376cf17b85e8151cb1e1bd/protocols/titrator_SOP.md)** Protocol for meauring total alkalinity in water samples, typically needed for studies that manipulate pCO2 or pH.
