
<html>
  <head>
    <title>genefish - crassostreome</title>
    <link rel="stylesheet" href="static/style.css" type="text/css" />
    <meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
  </head>
  <body>
    <div class="wiki" id="content_view" style="display: block;">


<table class="wiki_table">
    <tr>
        <td><h1 id="toc0"><a name="TJGR"></a><a href="http://gigadb.org/pacific_oyster/" target="_blank" rel="nofollow"><img src="files/Pacific%20oyster%20%C2%BB%20GigaDB-1.png" alt="Pacific oyster » GigaDB-1.png" title="Pacific oyster » GigaDB-1.png" style="height: 112px; width: 82px;" align="left" /></a>TJGR</h1>
 things just got real<br />
<span style="font-size: 80%;">Zhang, G; Fang, X; Guo, X; Li, L; Luo, R; Xu, F; Yang, P; Zhang, L; Wang, X; Qi, H; Zhu, Y; Yang, L; Huang, Z (2012) Genomic data from the Pacific oyster (Crassostrea gigas). GigaScience. <a class="wiki_link_ext" href="http://dx.doi.org/10.5524/100030" rel="nofollow">http://dx.doi.org/10.5524/100030</a></span><br />
<br />
<span style="font-size: 10px;"><strong>Other Resources:</strong> <a class="wiki_link_ext" href="http://oysterdb.cn/home.html" rel="nofollow">OysterDB</a> (Blast, Downloads) | Related manuscript: <a class="wiki_link_ext" href="http://dx.doi.org/10.1038/nature11413" rel="nofollow">doi:10.1038/nature11413</a></span><br />
<br />
<span style="color: #ffffff;">.......................................................................................................................</span><br />
</td>
        <td>[ http://genefish.wikispaces.com/page/xml/crassostreome?v=rss_2_0]<br />
</td>
    </tr>
</table>

<br />
<strong><span style="background-color: #ffffff;">oyster.v9_M</span></strong><br />


<table class="wiki_table">
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/armina/oyster.v9_M.fa" rel="nofollow">FASTA</a><br />
</td>
        <td><span style="background-color: #ffffff;">60 Sequences; 75 MB</span><br />
</td>
    </tr>
    <tr>
        <td>BSMAP<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/armina/BSMAP_output_trimmed_v9_M.sam" rel="nofollow">SAM </a><br />
</td>
        <td>BS MBD<br />
</td>
    </tr>
    <tr>
        <td>RNAseq<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/armina/RNAseq_solid0078_20091105_DH3.sam" rel="nofollow">SAM</a><br />
</td>
        <td>PROPS: DH<br />
</td>
    </tr>
    <tr>
        <td>RNAseq<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/armina/RNAseq_solid0078_20091105_BB3.sam" rel="nofollow">SAM</a><br />
</td>
        <td>PROPS: BBC<br />
</td>
    </tr>
    <tr>
        <td>RefMap<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/armina/solid0078_20110412_FRAG_BC_WHITE_WHITE_F3_QV_SB_MOTH_oyster_v9_M.sam" rel="nofollow">SAM</a><br />
</td>
        <td>MBD Library<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://128.95.149.81/cnidarian/BS_MBD_oysterv9_M_a.bed" rel="nofollow">BED</a><br />
</td>
        <td>BS MBD CG features (no scores)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://128.95.149.81/cnidarian/BS_MBD_methratio_oysterv9_M_d.BED" rel="nofollow">BED</a><br />
</td>
        <td>BS MBD CG features (with scores)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://128.95.149.81/cnidarian/BS_MBD_methratio_oysterv9_M_d_0_20.BED" rel="nofollow">BED</a><br />
</td>
        <td>BS MBD CG features (0-20%)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://128.95.149.81/cnidarian/BS_MBD_methratio_oysterv9_M_d_20_70.BED" rel="nofollow">BED</a><br />
</td>
        <td>BS MBD CG features (20-70%)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://128.95.149.81/cnidarian/BS_MBD_methratio_oysterv9_M_d_70_100.txt" rel="nofollow">BED</a><br />
</td>
        <td>BS MBD CG features (70-100%)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oyster_v9_M_repeats_phobos.gff" rel="nofollow">GFF</a><br />
</td>
        <td>Tandem Repeats [Phobos] (imperfect search)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oyster_v9_M_repeats_phobos_b.bed" rel="nofollow">BED</a><br />
</td>
        <td>Tandem Repeats (interval and ID only)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oyster_v9_M%5BComplement_to_CDS%5D.bed" rel="nofollow">BED</a><br />
</td>
        <td>Complement to CDS features<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oyster_v9_M_fuzznuc_CG.gff" rel="nofollow">GFF</a><br />
</td>
        <td>CG motifs<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oyster_v9_M_nonmeth_CG_250k.gff" rel="nofollow">GFF</a><br />
</td>
        <td>CG motifs non overlap w BS MBD CG (~250k)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oyster_v9_M_500k_CG_a.gff" rel="nofollow">GFF</a><br />
</td>
        <td>CG motifs with methylation status (~500k)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oyster_v9_M_500k_CG_b.bed" rel="nofollow">BED</a><br />
</td>
        <td>CG motifs(~500k) (interval and ID only)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oyster_v9_M_500k_100bp_flanking" rel="nofollow">BED</a><br />
</td>
        <td>50bp region (up and down) flanking CG motifs(~500k)<br />
</td>
    </tr>
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oyster_v9_M_500k_CG_flanking50.fasta" rel="nofollow">FASTA</a><br />
</td>
        <td>50bp region (up and down) flanking CG motifs(~500k)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oyster_v9_M_CGBS_repeat_overlap.gff" rel="nofollow">GFF</a><br />
</td>
        <td>CG motifs (of 500k) that overlap with Tandem Repeats<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/invREP_oyster_v9_M_blastnout_b" rel="nofollow">Blast Table</a><br />
</td>
        <td>Transposable Elements: RepBase inv Blast<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oyster_v9_M_RepBase_inv.bed" rel="nofollow">BED</a><br />
</td>
        <td>Transposable Elements: evalue 1E-10<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oyster_v9_M_RepBase_inv_b.bed" rel="nofollow">BED</a><br />
</td>
        <td>Transposable Elements: evalue 1E-10; single directionality<br />
</td>
    </tr>
    <tr>
        <td>Data<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oyster_v9_M_500k_CG_RepBase_JOIN.txt" rel="nofollow">Tab Text</a><br />
</td>
        <td>Interval Join: CG motifs(~500k) and TEs (RepBase)<br />
</td>
    </tr>
    <tr>
        <td>Data<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oyster_v9_M_500k_CG_Repeats_JOIN.txt" rel="nofollow">Tab Text</a><br />
</td>
        <td>Interval Join: CG motifs(~500k) and Repeats (Phobos)<br />
</td>
    </tr>
    <tr>
        <td>Data<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oyster_v9_M_500k_CG_Exons_JOIN.txt" rel="nofollow">Tab Text</a><br />
</td>
        <td>Interval Join: CG motifs(~500k) and exons (CDS)<br />
</td>
    </tr>
    <tr>
        <td>Data<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oyster_v9_M_500k_closest_nonoverlap_exons.txt" rel="nofollow">Tab Text</a><br />
</td>
        <td>Interval Join: CG motifs(~500k) and closest non overlapping exons<br />
</td>
    </tr>
    <tr>
        <td>Data<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oyster_v9_M_500k_CG_extragenic.txt" rel="nofollow">Tab Text</a><br />
</td>
        <td>Interval Join: CG motifs(~500k) not overlapping w/ mRNA (extragenic)<br />
</td>
    </tr>
</table>

<em><span style="background-color: #ffffff;">Description</span></em><span style="background-color: #ffffff;">: Genomic scaffolds longer that 1M bps</span><br />
<br />
<br />
<strong><span style="background-color: #ffffff;">oyster.v9_90</span></strong><br />


<table class="wiki_table">
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oyster.v9_90.fa" rel="nofollow">FASTA</a><br />
</td>
        <td><span style="background-color: #ffffff;">1670 Sequences; 488 MB</span><br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/TJGR_v9_90_ArrayDesign1.txt" rel="nofollow">Tab Text </a><br />
</td>
        <td>Tiling Array Design v1<br />
</td>
    </tr>
</table>

<em><span style="background-color: #ffffff;">Description</span></em><span style="background-color: #ffffff;">: Longest genomic scaffolds (1670; 14%) that cover over 90% of genome.</span><br />
<br />
<br />
<strong><span style="background-color: #ffffff;">oyster.v9_</span><span style="background-color: #ffffff; color: #800080;">proteome</span></strong><br />


<table class="wiki_table">
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="ftp://climb.genomics.cn/pub/10.5524/100001_101000/100030/gene_v9/oyster.v9.glean.final.rename.gff.pep.gz" rel="nofollow">FASTA</a><br />
</td>
        <td>[.gz] 28027 protein sequences (via gigadb.org)<br />
</td>
    </tr>
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/armina/oyster_v9_aa_format1.fasta" rel="nofollow">FASTA</a><br />
</td>
        <td>28027 protein sequences (local)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/armina/oyster_v9_aa_blastp_uniprot_b" rel="nofollow">Blast Table</a><br />
</td>
        <td>Blastp Swiss-Prot evalue: 10<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/armina/oyster_v9_aa_blastp_uniprot_c" rel="nofollow">Blast Table</a><br />
</td>
        <td>Blastp Swiss-Prot evalue: 1E-05<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/Estrogen_biosyn_proc_Blastout_oyster_protein.txt" rel="nofollow">Blast Table</a><br />
</td>
        <td>Blastp Estrogen Biosynthetic Process; 1E-20, 4 max_targets, (<a class="wiki_link_ext" href="https://www.evernote.com/shard/s10/sh/536e6c96-2e4f-4526-af60-8ec54fb5f64f/2c34ee6b1c59d9e3ab2fff832dbada40" rel="nofollow">details</a>)<br />
</td>
    </tr>
</table>

<em><span style="background-color: #ffffff;">Description</span></em><span style="background-color: #ffffff;">: Proteins</span><br />
<br />
<strong><span style="background-color: #ffffff;">oyster.v9_</span><span style="background-color: #ffffff; color: #800080;">genes</span></strong><br />


<table class="wiki_table">
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="ftp://climb.genomics.cn/pub/10.5524/100001_101000/100030/gene_v9/oyster.v9.glean.final.rename.gff.cds.gz" rel="nofollow">FASTA</a><br />
</td>
        <td>[.gz] 28027 gene (CDS only) sequences (via gigadb.org)<br />
</td>
    </tr>
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/armina/oyster_v9_gene.fasta" rel="nofollow">FASTA</a><br />
</td>
        <td>28027 gene (CDS only) sequences (local)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://128.95.149.81/cnidarian/oyster_v9_Gene_node2_blastout" rel="nofollow">Blast table</a><br />
</td>
        <td>Blastx Swiss-Prot evalue: 1E-05<br />
</td>
    </tr>
    <tr>
        <td>Data<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oyster_v9_M_gene_exon_number.txt" rel="nofollow">Tab Text</a><br />
</td>
        <td>Number of exons per gene<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/TJGR_Gene_SPID_evalue.txt" rel="nofollow">Tab Text</a><br />
</td>
        <td>Corresponding SPID and evalues<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/TJGR_Gene_SPID_evalue_Description.txt" rel="nofollow">Tab Text</a><br />
</td>
        <td>Corresponding SPID, evalues, and descriptions<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/TJGR_Gene_SPID_evalue_GO_bigfile.txt" rel="nofollow">Tab Text</a><br />
</td>
        <td>Corresponding SPID, evalues, and GO# (using <a class="wiki_link_ext" href="http://eagle.fish.washington.edu/whale/SPID_and_GO_number.tabular" rel="nofollow">recent GO file</a>)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/TJGR_Gene_GO_GOslim.txt" rel="nofollow">Tab Text</a><br />
</td>
        <td>Corresponding SPID, evalues, and GO and GOslim<br />
</td>
    </tr>
</table>

<em><span style="background-color: #ffffff;">Description</span></em><span style="background-color: #ffffff;">: Coding sequence</span><br />
<br />
<br />
<br />
<strong><span style="background-color: #ffffff;">oyster.v9</span></strong><br />


<table class="wiki_table">
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="ftp://climb.genomics.cn/pub/10.5524/100001_101000/100030/gene_v9/oyster.v9.fa.gz" rel="nofollow">FASTA</a><br />
</td>
        <td>[.gz] 11969 Sequences (via gigadb.org)<br />
</td>
    </tr>
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/armina/oyster.v9.fa.gz" rel="nofollow">FASTA</a><br />
</td>
        <td><span style="background-color: #ffffff;">[.gz] 11969 Sequences (local)</span><br />
</td>
    </tr>
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/armina/oyster.v9.fa" rel="nofollow">FASTA</a><br />
</td>
        <td><span style="background-color: #ffffff;">11969 Sequences (local) 560MB</span><br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="ftp://climb.genomics.cn/pub/10.5524/100001_101000/100030/gene_v9/oyster.v9.glean.final.rename.gff.gz" rel="nofollow">GFF</a><br />
</td>
        <td>[.gz] gene features (via gigadb.org)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/armina/oyster.v9.glean.final.rename.gff" rel="nofollow">GFF</a><br />
</td>
        <td>gene features (CDS and mRNA)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/armina/oyster.v9.glean.final.rename.CDS.gff" rel="nofollow">GFF</a><br />
</td>
        <td>gene features (CDS only)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/TJGR_oyster_v9_CDS.bed" rel="nofollow">BED</a><br />
</td>
        <td>gene features (CDS only) (interval and ID)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/armina/oyster.v9.glean.final.rename.mRNA.gff" rel="nofollow">GFF</a><br />
</td>
        <td>gene features (mRNA only)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/oysterv9_mRNA_BED.txt" rel="nofollow">BED</a><br />
</td>
        <td>gene features (mRNA only)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/TJGR_genes_v9_promoter_5p1000.gff" rel="nofollow">GFF</a><br />
</td>
        <td>promoter region (1000bp 5' of mRNA)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/TJGR_genes_v9_cellad_sigtran_ccsig.gff" rel="nofollow">GFF</a><br />
</td>
        <td>mRNA GOslim = cell adhesion or signal transduction or cell-cell signaling<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/TJGR_genes_v9_DNAmet_RNAmet_promet.gff" rel="nofollow">GFF</a><br />
</td>
        <td>mRNA GOslim = DNA metabolism or RNA metabolism or protein metabolism<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/DESeq_GillvGonad_p05.gff" rel="nofollow">GFF</a><br />
</td>
        <td>DESeq (p&lt;0.05) Gill v Male Gonad RNA-seq from Zhang et al<br />
</td>
    </tr>
</table>

<em><span style="background-color: #ffffff;">Description</span></em><span style="background-color: #ffffff;">: </span>Zhang, G; Fang, X; Guo, X; Li, L; Luo, R; Xu, F; Yang, P; Zhang, L; Wang, X; Qi, H; Zhu, Y; Yang, L; Huang, Z (2012) Genomic data from the Pacific oyster (<em>Crassostrea gigas</em>). GigaScience. <a class="wiki_link_ext" href="http://dx.doi.org/10.5524/100030" rel="nofollow">http://dx.doi.org/10.5524/100030</a><br />
<br />
<br />
<br />
<br />


<table class="wiki_table">
    <tr>
        <td><h1 id="toc1"><a name="TGAGA (alpha)"></a><img src="https://img.skitch.com/20120227-buriq24ytfwyrjsh4ut9qw4nhb.png" alt="external image 20120227-buriq24ytfwyrjsh4ut9qw4nhb.png" title="external image 20120227-buriq24ytfwyrjsh4ut9qw4nhb.png" align="left" />TGAGA (alpha)</h1>
 towards getting a gigas assembly (alpha)<br />
<span style="color: #ffffff;">................................................................................................................................................</span><br />
</td>
        <td><br />
</td>
    </tr>
</table>

<br />
<span style="display: block; text-align: right;"><br />
</span><br />
<strong>cgigas_alpha_v0.4.0</strong><br />


<table class="wiki_table">
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/ballyhoo/cgigas_alpha_v040.fa" rel="nofollow">FASTA</a><br />
</td>
        <td><br />
</td>
        <td>230,270 Sequences<br />
</td>
    </tr>
</table>

<em>Description</em>: Independent assembly of 19 fosmids. Sequences were analyzed using CD-HIT EST to obtain non-redundant dataset.<br />
Combination of assemblies v0.2.0 and v0.3.1.<br />
<br />
<strong>cgigas_alpha_v0.3.2</strong><br />


<table class="wiki_table">
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/cgigas_alpha_v032.fa" rel="nofollow">FASTA</a><br />
</td>
        <td>272 Sequences<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/Annotation1_cg_v032.gff" rel="nofollow">GFF v1 (gff)</a><br />
</td>
        <td>blastn Sigenae8<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/Annotation1_cg_v032.bed" rel="nofollow">BED v1 (bed)</a><br />
</td>
        <td>blastn Sigenae8<br />
</td>
    </tr>
    <tr>
        <td>Visual<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/ballyhoo/Anno_v032Table.htm" rel="nofollow">Blast Directionality (htm)</a><br />
</td>
        <td>based on BED v1<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/Annotation2_cg_v032.gff" rel="nofollow">GFF v2 (gff)</a><br />
</td>
        <td>blastn Sigenae8 (includes gene description)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/ballyhoo/Annotation3_cg_v032.gff" rel="nofollow">GFF v3 (gff)</a><br />
</td>
        <td>blastn Sigenae8 GOslim= cell adhesion or signal transduction or cell-cell signaling<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/ballyhoo/Annotation3_cg_v032.bed" rel="nofollow">GFF v3 (bed)</a><br />
</td>
        <td>blastn Sigenae8 GOslim= cell adhesion or signal transduction or cell-cell signaling<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/ballyhoo/Annotation4_cg_v032.gff" rel="nofollow">GFF v4 (gff)</a><br />
</td>
        <td>blastn Sigenae8 GOslim= DNA metabolism or RNA metabolism or protein metabolism<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/ballyhoo/Annotation4_cg_v032.bed" rel="nofollow">GFF v4 (bed)</a><br />
</td>
        <td>blastn Sigenae8 GOslim= DNA metabolism or RNA metabolism or protein metabolism<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a href="files/MicroSatellite_GFF_Final1.txt">MicroSatellite (GFF3</a>)<br />
</td>
        <td>MicroSatellite Regions (HP)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a href="files/familyGFF.txt">Repeat_Regions(GFF3)</a><br />
</td>
        <td>Regions that are repeated 10 or more times(HP)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a href="files/nestsRemoved3.txt">Repeat_Regions v1 (GFF3)</a><br />
</td>
        <td>Regions that are repeated 10 or more times, derivative regions omitted (HP)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a href="files/finalNest.txt">Repeat_Regions v2 (GFF3)</a><br />
</td>
        <td>Fixed orientation and score issues (HP)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a href="files/finalNest.names.txt">Repeat_Regions v3 (GFF3)</a><br />
</td>
        <td>Identified some Retrotransposon regions from Repbase (HP)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a href="files/repeat_regions_beta.bed">Repeat_Regions(BED)</a><br />
</td>
        <td>Regions that are repeated 10 or more times(HP)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a href="files/cleanNest.bed">Repeat_Regions v1 (BED)</a><br />
</td>
        <td>Regions that are repeated 10 or more times, derivative regions omitted(HP)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/ballyhoo/Annotation5_cg_v032_tandem.gff" rel="nofollow">GFF v5 (gff)</a><br />
</td>
        <td>Tandem repeats (Geneious:Phobos)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/ballyhoo/Annotation6_cg_v032_TF.gff" rel="nofollow">GFF v6 (gff)</a><br />
</td>
        <td>Transcription Factors (Geneious)<br />
</td>
    </tr>
    <tr>
        <td>Alignment<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/RefMap_MBD_meth_v032.bed" rel="nofollow">RefMap MBD_meth (bed)</a><br />
</td>
        <td>gill tissue<br />
</td>
    </tr>
    <tr>
        <td>Alignment<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/RefMap_MBD_UNmeth_v032.bed" rel="nofollow">RefMap MBD_unmeth (bed)</a><br />
</td>
        <td>gill tissue<br />
</td>
    </tr>
    <tr>
        <td>Alignment<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/RNAseq_BB3_v032.bed" rel="nofollow">RNAseq BB3 (bed)</a><br />
</td>
        <td>gill tissue<br />
</td>
    </tr>
    <tr>
        <td>Alignment<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/RNAseq_DH3_v032.bed" rel="nofollow">RNAseq DH3 (bed)</a><br />
</td>
        <td>gill tissue<br />
</td>
    </tr>
    <tr>
        <td>Alignment<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/ballyhoo/SRP002286_larvae_cgigas_v032.bed" rel="nofollow">RefMap larvae_SRP002286 (bed)</a><br />
</td>
        <td>larvae<br />
</td>
    </tr>
    <tr>
        <td>Alignment<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/ballyhoo/RefMap_SRA454_SRP004696_v032.bed" rel="nofollow">RefMap larvae_SRP004696 (bed)</a><br />
</td>
        <td>Two 454 SRA files - larvae<br />
</td>
    </tr>
    <tr>
        <td>Motif<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/ballyhoo/cg_motif_cgigas_v032.gff" rel="nofollow">CG motifs (gff)</a><br />
</td>
        <td><br />
</td>
    </tr>
    <tr>
        <td>BSMAP<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/ballyhoo/bsmap_all_cgigas_v032.gff" rel="nofollow">BSMAP_MBD_all (gff)</a><br />
</td>
        <td>Bisulfite-seq 10x cov ; all<br />
</td>
    </tr>
    <tr>
        <td>BSMAP<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/ballyhoo/bsmap_none_cgigas_v032.gff" rel="nofollow">BSMAP_MBD_0 (gff)</a><br />
</td>
        <td>Bisulfite-seq 10x cov ; 0% methylated<br />
</td>
    </tr>
    <tr>
        <td>BSMAP<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/ballyhoo/bsmap_0_10_cgigas_v032.gff" rel="nofollow">BSMAP_MBD_1_9 (gff)</a><br />
</td>
        <td>Bisulfite-seq 10x cov ; 1-9% methylated<br />
</td>
    </tr>
    <tr>
        <td>BSMAP<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/ballyhoo/bsmap_10_50_cgigas_v032.gff" rel="nofollow">BSMAP_MBD_10_49 (gff)</a><br />
</td>
        <td>Bisulfite-seq 10x cov ; 10-49% methylated<br />
</td>
    </tr>
    <tr>
        <td>BSMAP<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/ballyhoo/bsmap_50_90_cgigas_v032.gff" rel="nofollow">BSMAP_MBD_50_89 (gff)</a><br />
</td>
        <td>Bisulfite-seq 10x cov ; 50-89% methylated<br />
</td>
    </tr>
    <tr>
        <td>BSMAP<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/ballyhoo/bsmap_90_100_cgigas_v032.gff" rel="nofollow">BSMAP_MBD_90_100 (gff)</a><br />
</td>
        <td>Bisulfite-seq 10x cov ; 90-100% methylated<br />
</td>
    </tr>
    <tr>
        <td>Visual<br />
</td>
        <td><a class="wiki_link_ext" href="https://main.g2.bx.psu.edu/u/sroberts320/v/cgigasv032-bs" rel="nofollow">Track Visualization BS (Galaxy) </a><br />
</td>
        <td>Tracks include: exon annotation, RNA-seq, select GO terms, CpG methylation<br />
</td>
    </tr>
    <tr>
        <td>Visual<br />
</td>
        <td><a class="wiki_link_ext" href="http://main.g2.bx.psu.edu/u/sroberts320/v/454-on-cgigasv032" rel="nofollow">Track Visualization 454 (Galaxy) </a><br />
</td>
        <td>Tracks include: RefMap larvae_SRP004696 (bed)<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/ballyhoo/First_Half_-_Convert_Wiggle_to_BigWig_on_data_146.bigwig" rel="nofollow">bigwig 1</a><br />
</td>
        <td><br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/ballyhoo/Second_Half_-_Convert_Wiggle_to_BigWig_on_data_147.bigwig" rel="nofollow">bigwig 2</a><br />
</td>
        <td><br />
</td>
    </tr>
</table>

<em>Description</em>: Independent assembly of 10 fosmids. Sequences were analyzed using CD-HIT EST to obtain non-redundant dataset. Sequences greater than 20k bp retained. Cgigas BAC clones from NCBI (60) reduced to 53 clusters. Plus 12 select sequences with known genomic structure.<br />
<br />
<br />
<br />
<strong>cgigas_alpha_v0.3.1</strong><br />


<table class="wiki_table">
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/cgigas_alpha_v031.fa" rel="nofollow">FASTA</a><br />
</td>
        <td>500MB<br />
</td>
        <td>203,216 Sequences<br />
</td>
    </tr>
</table>

<em>Description</em>: Independent assembly of 10 fosmids. Sequences were analyzed using CD-HIT EST to obtain non-redundant dataset.<br />
<br />
<br />
<strong>cgigas_alpha_v0.3.0</strong><br />


<table class="wiki_table">
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/cgigas_alpha_v030.fa" rel="nofollow">FASTA</a><br />
</td>
        <td>12.3MB<br />
</td>
        <td>272 Sequences<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/Annotation1_cg_v030.gff" rel="nofollow">GFF v1 (gff)</a><br />
</td>
        <td><br />
</td>
        <td>blastn Sigenae8<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/Annotation1_cg_v030.bed" rel="nofollow">BED v1 (bed)</a><br />
</td>
        <td><br />
</td>
        <td>blastn Sigenae8<br />
</td>
    </tr>
    <tr>
        <td>Alignment<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/MBD_meth_refmap_v030.sam" rel="nofollow">MBD-meth reads (sam)</a><br />
</td>
        <td><br />
</td>
        <td><br />
</td>
    </tr>
    <tr>
        <td>Alignment<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/MBD_UNmeth_refmap_v030.sam" rel="nofollow">MBD-ummeth reads (sam)</a><br />
</td>
        <td><br />
</td>
        <td><br />
</td>
    </tr>
    <tr>
        <td>Alignment<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/BB3_RNAseq_v030.sam" rel="nofollow">RNAseq BB3 (sam) </a><br />
</td>
        <td><br />
</td>
        <td><br />
</td>
    </tr>
    <tr>
        <td>Alignment<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/DH3_RNAseq_v030.sam" rel="nofollow">RNAseq DH3 (sam)</a><br />
</td>
        <td><br />
</td>
        <td><br />
</td>
    </tr>
</table>

<em>Description</em>: Independent assembly of 10 fosmids. Consensus sequences &gt;20,000bp along with publicly available BAC sequences were analyzed using CD-HIT EST to obtain non-redundant dataset. Plus 12 select sequences with known genomic structure.<br />
<br />
<br />
<strong>cgigas_alpha_v0.2.0</strong><br />


<table class="wiki_table">
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="http://main.g2.bx.psu.edu/datasets/83ae21f9d999536d/display?username=sr320&amp;to_ext=fasta&amp;slug=repository" rel="nofollow">FASTA</a><br />
</td>
        <td>615MB<br />
</td>
        <td>205,903 Sequences<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><br />
</td>
        <td><br />
</td>
        <td><br />
</td>
    </tr>
    <tr>
        <td><br />
</td>
        <td><br />
</td>
        <td><br />
</td>
        <td><br />
</td>
    </tr>
</table>

<em>Description</em>: Independent assembly of nine fosmids. Consensus sequences were analyzed using CD-HIT EST to obtain non-redundant dataset.<br />
<br />
<br />
<strong>cgigas_alpha_v0.1.3</strong><br />


<table class="wiki_table">
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/cgigas_alpha_v013.fa" rel="nofollow">FASTA</a><br />
</td>
        <td>6MB<br />
</td>
        <td>262 Sequences<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><br />
</td>
        <td><br />
</td>
        <td><br />
</td>
    </tr>
    <tr>
        <td><br />
</td>
        <td><br />
</td>
        <td><br />
</td>
        <td><br />
</td>
    </tr>
</table>

<em>Description</em>: Independent assembly of nine fosmids. Consensus sequences with greater than 20x average coverage and longer than 5000 bp were analyzed using CD-HIT EST to obtain non-redundant dataset. Remaining sequences greater than 20,000 bp are included. Plus 12 select sequences with known genomic structure.<br />
<br />
<br />
<strong>cgigas_alpha_v0.1.2</strong><br />


<table class="wiki_table">
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/cgigas_alpha_v012.fa" rel="nofollow">FASTA</a><br />
</td>
        <td>215MB<br />
</td>
        <td>28,538 Sequences<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/blastn_sig8_v012.out" rel="nofollow">BLASTn Sigenae v8 (tsv)</a><br />
</td>
        <td>800MB<br />
</td>
        <td>*rev<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/Annotation_cgigas_v012_sigenae8_1.gff" rel="nofollow">GFF v1 (gff)</a><br />
</td>
        <td>9MB<br />
</td>
        <td><br />
</td>
    </tr>
</table>

<em>Description</em>: Independent assembly of nine fosmids. Consensus sequences with greater than 20x average coverage and longer than 5000 bp were analyzed using CD-HIT EST to obtain non-redundant dataset. Plus 12 select sequences with known genomic structure.<br />
<br />
<br />
<strong>cgigas_alpha_v0.1.1</strong><br />


<table class="wiki_table">
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="http://main.g2.bx.psu.edu/datasets/d4a56e6992f8319e/display?username=sr320&amp;to_ext=fasta&amp;slug=repository" rel="nofollow">FASTA</a><br />
</td>
        <td>215MB<br />
</td>
        <td>28,526 Sequences<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><br />
</td>
        <td><br />
</td>
        <td><br />
</td>
    </tr>
    <tr>
        <td><br />
</td>
        <td><br />
</td>
        <td><br />
</td>
        <td><br />
</td>
    </tr>
</table>

<em>Description</em>: Independent assembly of nine fosmids. Consensus sequences with greater than 20x average coverage and longer than 5000 bp were analyzed using CD-HIT EST to obtain non-redundant dataset.<br />
<br />
<br />
<strong>cgigas_alpha_v0.1.0</strong><br />


<table class="wiki_table">
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="http://main.g2.bx.psu.edu/datasets/c9ae3ae663bab4b2/display?username=sr320&amp;to_ext=fasta&amp;slug=repository" rel="nofollow">FASTA</a><br />
</td>
        <td>5.8MB<br />
</td>
        <td>250 Sequences<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://dl.dropbox.com/u/115356/docs/annotable/cg0.1.0_Blastnout_sigenaev8.csv" rel="nofollow">BLASTn Sigenae v8 (csv)</a><br />
</td>
        <td><br />
</td>
        <td>top hit<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="https://www.evernote.com/shard/s10/res/641c0adb-f226-4adc-aeb4-83e01f07dfa7/Combined_fosmids_cd_hit_mod_20000_CgSig8.txt" rel="nofollow">SW Sigenae v8 (tsv)</a><br />
</td>
        <td><br />
</td>
        <td>top hit<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/Annotation0_cg_v010.gff" rel="nofollow">GFF v0 (gff)</a><br />
</td>
        <td><br />
</td>
        <td>12,582 total hits<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/Annotation1_cg_v010.gff" rel="nofollow">GFF v1 (gff)</a><br />
</td>
        <td><br />
</td>
        <td>3891 hits, gene names<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/Annotation2_cg_v010.gff" rel="nofollow">GFF v2 (gff)</a><br />
</td>
        <td><br />
</td>
        <td>1105 hits, used rev blast<br />
</td>
    </tr>
</table>

<em>Description</em>: Independent assembly of nine fosmids. Consensus sequences with greater than 20x average coverage and longer than 5000 bp were analyzed using CD-HIT EST to obtain non-redundant dataset. Remaining sequences greater than 20,000 bp are included.<br />
<br />
<br />
<strong>cgigas_PREalpha_v0.2.0</strong><br />


<table class="wiki_table">
    <tr>
        <td>Sequence<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/cgigas_prealpha_v020.fa" rel="nofollow">FASTA</a><br />
</td>
        <td><br />
</td>
        <td>12 sequences<br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/Annotations_1_PREalpha_v020.gff" rel="nofollow">GFF v1 (gff)</a><br />
</td>
        <td><br />
</td>
        <td><br />
</td>
    </tr>
    <tr>
        <td>Annotation<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/Annotations_2_PREalpha_v020.gff" rel="nofollow">GFF v2 (gff) </a><br />
</td>
        <td><br />
</td>
        <td>Exons only<br />
</td>
    </tr>
    <tr>
        <td>Alignment<br />
</td>
        <td><a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/filtered_Unlabeled_NoIndex_L003_R1%20trimmed%20mapping.sam" rel="nofollow">MBD-BS reads (sam)</a><br />
</td>
        <td><br />
</td>
        <td><br />
</td>
    </tr>
</table>

<br />
<br />
*<em>BLAST option:</em> <a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/blast/blast.html" rel="nofollow" target="_blank">blastn</a>, <a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/blast/megablast.html" rel="nofollow" target="_blank">megablast</a> select database from menu<br />
<br />
<hr />
<h1 id="toc2"><a name="Related Resources"></a><img src="https://img.skitch.com/20120227-f95u9h9wt2gqwihpusy1qccqrx.png" alt="external image 20120227-f95u9h9wt2gqwihpusy1qccqrx.png" title="external image 20120227-f95u9h9wt2gqwihpusy1qccqrx.png" align="left" />Related Resources</h1>
 <span style="color: #aaaaaa;">various other genomic resources for the oyster.</span><br />
<br />
<br />
<br />
<br />
<hr />
<br />
<h2 id="toc3"><a name="Related Resources-GigasBase"></a>GigasBase</h2>
 <h2 id="toc4"><a name="Related Resources-Public Sigenae Contig Browser: Oyster"></a><span style="font-size: 14px; line-height: 21px;"><a class="wiki_link_ext" href="http://public-contigbrowser.sigenae.org:9090/Crassostrea_gigas/index.html" rel="nofollow">Public Sigenae Contig Browser: Oyster</a></span></h2>
 The Oyster EST contig browser aims to produce and maintain an automatic annotation of Oyster EST libraries. This database <u><span style="color: #318c40;"><a class="wiki_link_ext" href="http://www.biomedcentral.com/1471-2164/10/341" rel="nofollow">GigasDatabase</a></span></u> was initiated within the frame of the <u><span style="color: #318c40;"><a class="wiki_link_ext" href="http://www.sigenae.org/aquafirst" rel="nofollow">AquaFirst</a></span></u> European project, it now gathers EST sequences produced by a <u><span style="color: #318c40;"><a class="wiki_link_ext" href="http://www.marine-genomics-europe.org/" rel="nofollow">Marine Genomics Europe</a></span></u> project (GOCE-CT-2004-505403) and a <u><span style="color: #318c40;"><a class="wiki_link_ext" href="http://www.genoscope.cns.fr/spip/crassostrea-gigas-an-economically.html" rel="nofollow">Genoscope project</a></span></u>. <u><span style="color: #318c40;"><a class="wiki_link_ext" href="http://www.biomedcentral.com/1471-2164/10/341" rel="nofollow">GigasDatabase</a></span></u> is regularly updated in the context of the ANR project &quot;Gametogenes&quot; (ANR-08-GENM-041).<br />
<br />
<br />
<h4 id="toc5"><a name="Related Resources-Public Sigenae Contig Browser: Oyster--GigasDatabase Assemblies"></a>GigasDatabase Assemblies</h4>
 Version 8<br />
<a class="wiki_link_ext" href="http://aquacul4.fish.washington.edu/~steven/filefish/cgigas_all_contigs_v8.fa" rel="nofollow">cgigas_all_contigs_v8.fa</a><br />
<a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/Cg_Sigenae8_besthit.tsv" rel="nofollow">cgigas_all_contigs_v8_BESTHIT.tsv</a><br />
<a class="wiki_link_ext" href="http://eagle.fish.washington.edu/cnidarian/Cg_Sigenae8_ontology.txt" rel="nofollow">cgigas_all_contigs_v8_ontology.tsv</a><br />
<br />
<br />
<h4 id="toc6"><a name="Related Resources-Public Sigenae Contig Browser: Oyster--Archived GigasDatabase Assemblies"></a>Archived GigasDatabase Assemblies</h4>
 Version 6<br />
<a class="wiki_link_ext" href="http://dl.dropbox.com/u/115356/docs/annotable/Roberts_Sigenae6_transcriptome.fa" rel="nofollow">Sigenae_v6_assembly.fa</a><br />
<a class="wiki_link_ext" href="http://dl.dropbox.com/u/115356/docs/annotable/Sigenaev6_hitonly.xls" rel="nofollow">Sigenae _v6_SPhits.xls</a><br />
<br />
<hr />
<br />
<br />
<h2 id="toc7"><a name="Related Resources-NCBI"></a>NCBI</h2>
 <a class="wiki_link_ext" href="http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&amp;id=29159&amp;lvl=3&amp;lin=f&amp;keep=1&amp;srchmode=1&amp;unlock" rel="nofollow">Crassostrea gigas Entrez Records</a><br />
<a class="wiki_link_ext" href="http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&amp;id=6565&amp;lvl=3&amp;lin=f&amp;keep=1&amp;srchmode=1&amp;unlock" rel="nofollow">Crassostrea virginica Entrez Records</a><br />
<br />
<h3 id="toc8"><a name="Related Resources-NCBI-NCBI: SRA"></a>NCBI: SRA</h3>
 <a class="wiki_link_ext" href="http://www.ncbi.nlm.nih.gov/sites/entrez?db=sra&amp;cmd=Search&amp;dopt=DocSum&amp;term=txid29159%5BOrganism:noexp%5D" rel="nofollow">Crassostrea gigas</a><br />
<a class="wiki_link_ext" href="http://dl.dropbox.com/u/115356/docs/annotable/Cgigas454_all_reads_trimmed.fa" rel="nofollow"> FASTA export of quality trimmed reads</a> (export date 02/10/11)<br />
<br />
<h3 id="toc9"><a name="Related Resources-NCBI-Misc"></a>Misc</h3>
 <a class="wiki_link_ext" href="http://dl.dropbox.com/u/115356/docs/annotable/Roberts_Cgigas_BAConly.fa" rel="nofollow">Cgigas_BAConly.fa</a> (export date 01/31/11)<br />
<a class="wiki_link_ext" href="http://dl.dropbox.com/u/115356/docs/annotable/Roberts_Cgigas_genomic_NCBI.fa" rel="nofollow">Cgigas_genomic_NCBI.fa</a> (export date 01/31/11)<br />
<br />
<h3 id="toc10"><a name="Related Resources-NCBI-Roberts Lab Submissions"></a><span style="font-size: 1.1em; margin: 0px; padding: 5px 0px 0px;"><strong>Roberts Lab Submissions</strong></span></h3>
 <a class="wiki_link_ext" href="http://www.ncbi.nlm.nih.gov/nuccore?term=Roberts%5BAuthor%5D%20AND%20Crassostrea%5Borgn%5D" rel="nofollow">Crassostrea Nucleotide Database entries</a><br />
<a class="wiki_link_ext" href="http://www.ncbi.nlm.nih.gov/nucest?term=Roberts%5BAuthor%5D%20AND%20Crassostrea%5Borgn%5D" rel="nofollow">Crassostrea gigas EST Database entries</a><br />
<br />
<br />
<hr />
<br />
<br />
<br />
<hr />
<br />
<span style="display: block; text-align: left;"><a title="customizable counter" href="http://statcounter.com/free-hit-counter/" target="_blank" rel="nofollow"><img src="http://c.statcounter.com/7786424/0/14c1e656/1/" alt="customizable counter" style="border:none;" /></a></span><span style="display: block; text-align: left;"><script type="text/javascript">
//<![CDATA[

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-306430-13']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

//]]>
</script><br />
</span>
    </div>
  </body>
</html>