
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../Computing_GMGI/">
      
      
        <link rel="next" href="../Computing_Begin_Resources/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.37">
    
    
      
        <title>Northeastern University - GMGI Fisheries Handbook</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.8c3ca2c6.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="blue" data-md-color-accent="blue">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#northeastern-university-computing-resources" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="GMGI Fisheries Handbook" class="md-header__button md-logo" aria-label="GMGI Fisheries Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            GMGI Fisheries Handbook
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Northeastern University
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="blue" data-md-color-accent="blue"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 6H7c-3.31 0-6 2.69-6 6s2.69 6 6 6h10c3.31 0 6-2.69 6-6s-2.69-6-6-6m0 10H7c-2.21 0-4-1.79-4-4s1.79-4 4-4h10c2.21 0 4 1.79 4 4s-1.79 4-4 4M7 9c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="blue" data-md-color-accent="blue"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/GMGI-Fisheries/resources" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="GMGI Fisheries Handbook" class="md-nav__button md-logo" aria-label="GMGI Fisheries Handbook" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"/></svg>

    </a>
    GMGI Fisheries Handbook
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/GMGI-Fisheries/resources" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Computing Hardware
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Computing Hardware
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../Computing_GMGI/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    GMGI
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Northeastern University
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Northeastern University
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#logging-in" class="md-nav__link">
    <span class="md-ellipsis">
      Logging in
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#server-structure" class="md-nav__link">
    <span class="md-ellipsis">
      Server structure
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#storage-rules" class="md-nav__link">
    <span class="md-ellipsis">
      Storage rules
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nu-contacts-and-research-computing-help" class="md-nav__link">
    <span class="md-ellipsis">
      NU Contacts and Research Computing Help
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-a-bioinformatic-script" class="md-nav__link">
    <span class="md-ellipsis">
      Running a bioinformatic script
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#packages-and-modules" class="md-nav__link">
    <span class="md-ellipsis">
      Packages and modules
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../Computing_Begin_Resources/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Computing Beginner Resources
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../tbd.md" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Data Management
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../Data_To_Server/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Data Transfer
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    eDNA: metabarcoding
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            eDNA: metabarcoding
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../eDNA/01-Metabarcoding%20ampliseq%2012S%20Riaz/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    01-ampliseq-12S-Riaz
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../eDNA/01-Metabarcoding%20ampliseq%2012S%20MiFish/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    01-ampliseq-12S-MiFish
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../eDNA/02-datatable_prep_12S_Riaz/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    02-dfprep-12S-Riaz
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../eDNA/03-data_quality/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    03-data-quality
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../eDNA/04-relative_abundance_heatmap/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    04-relative-abund-heatmap
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    eDNA: qPCR
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            eDNA: qPCR
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../tbd.md" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    tbd
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_8" >
        
          
          <label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    PopGen
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_8">
            <span class="md-nav__icon md-icon"></span>
            PopGen
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../PopGen/eDNA_01/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    01-
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_9" >
        
          
          <label class="md-nav__link" for="__nav_9" id="__nav_9_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Epigenetic Aging
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_9_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_9">
            <span class="md-nav__icon md-icon"></span>
            Epigenetic Aging
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../EpiAge/eDNA_01/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    01-
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_10" >
        
          
          <label class="md-nav__link" for="__nav_10" id="__nav_10_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Genome Assembly
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_10_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_10">
            <span class="md-nav__icon md-icon"></span>
            Genome Assembly
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../tbd.md" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    coming soon!
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_11" >
        
          
          <label class="md-nav__link" for="__nav_11" id="__nav_11_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Laboratory Protocols
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_11_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_11">
            <span class="md-nav__icon md-icon"></span>
            Laboratory Protocols
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_11_1" >
        
          
          <label class="md-nav__link" for="__nav_11_1" id="__nav_11_1_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    eDNA metabarcoding
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_11_1_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_11_1">
            <span class="md-nav__icon md-icon"></span>
            eDNA metabarcoding
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../eDNA/eDNA_01.md" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    01-
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#logging-in" class="md-nav__link">
    <span class="md-ellipsis">
      Logging in
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#server-structure" class="md-nav__link">
    <span class="md-ellipsis">
      Server structure
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#storage-rules" class="md-nav__link">
    <span class="md-ellipsis">
      Storage rules
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nu-contacts-and-research-computing-help" class="md-nav__link">
    <span class="md-ellipsis">
      NU Contacts and Research Computing Help
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-a-bioinformatic-script" class="md-nav__link">
    <span class="md-ellipsis">
      Running a bioinformatic script
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#packages-and-modules" class="md-nav__link">
    <span class="md-ellipsis">
      Packages and modules
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
    <a href="https://github.com/GMGI-Fisheries/resources/blob/master/docs/Computing_NU.md" title="Edit this page" class="md-content__button md-icon">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75z"/></svg>
    </a>
  
  


<h1 id="northeastern-university-computing-resources">Northeastern University Computing Resources</h1>
<p>A high-performance computing resource for the Northeastern research community, the Discovery cluster is located in the <a href="https://www.mghpcc.org/">Massachusetts Green High Performance Computing Center</a> in Holyoke, Massachusetts.</p>
<p>The Discovery cluster provides Northeastern researchers with access to more than 45,000 CPU cores and more than 400 GPUs. Connected to the university network over 10 Gbps Ethernet (GbE) for high-speed data transfer, Discovery provides 5 PB of available storage on a high-performance GPFS parallel file system. Compute nodes are connected with either 10 GbE or a high-performance HDR100 InfiniBand (IB) interconnect running at 200 Gbps, supporting all types and scales of computational workloads.</p>
<p>As GMGI's researchers, we have access to Northeastern's HPC resources through an MOU established in Fall 2023. </p>
<p>Read the HPC resource documentation prior to getting started: <a href="https://rc-docs.northeastern.edu/en/latest/welcome/index.html">Research Computing - NURC RTD (northeastern.edu)</a>.</p>
<p><img alt="" src="https://github.com/GMGI-Fisheries/resources/blob/master/img/NU_computing_resources.png?raw=true" /></p>
<h2 id="logging-in">Logging in</h2>
<p>Before creating an account with the NU Discovery Cluster, claim your Northeastern email and sponsored account.</p>
<ol>
<li>Log into (with the northeastern email and pw previously claimed): <a href="https://service.northeastern.edu/tech?id=tech_index_home">Home - Northeastern Tech Service Portal</a>.  </li>
<li>Navigate to <a href="https://service.northeastern.edu/tech?id=sc_category&amp;sys_id=43a3aef7db45cdd0ca10819b13961998">High Performance Computing - Northeastern Tech Service Portal</a>.    </li>
<li>Request an account (<a href="https://service.northeastern.edu/tech?id=sc_cat_item&amp;sys_id=0ae24596db535fc075892f17d496199c">Research Computing Access Request - Northeastern Tech Service Portal</a>). Fill out the form with your Northeastern email, and following options:  </li>
<li>Select: I do not have access to Discovery. I am requesting a new account.  </li>
<li>Affiliation with Northeastern University: Visiting Researcher (Greg says this answer doesn't really matter).   </li>
<li>University Sponsor: Geoffrey Trusell  </li>
<li>Gaussian: No (This is a specific program, if you don't know what it is, you don't need it).  </li>
<li>Select: By clicking here, I acknowledge that I have read and agree to the following.  </li>
<li>This triggers an email to Geoff Trusell to approve your account. Send an email to Geoff letting him know that you are activating your account that needs his approval.  </li>
<li>Once Geoff approves the account sponsorship, then Greg and the computing team will finish setting up your account.</li>
</ol>
<p>You can operate on the Discovery Cluster in two ways:<br />
1. via Linux operating system on your computer or ssh client<br />
2. NU's Open On Demand (<a href="https://rc-docs.northeastern.edu/en/latest/using-ood/index.html">Open OnDemand (OOD) - RC RTD (northeastern.edu)</a>) GUI. <a href="https://rc-docs.northeastern.edu/en/latest/using-ood/accessingood.html#access-ood">Accessing Open OnDemand - RC RTD (northeastern.edu)</a>. With OOD interface, you can access plug-ins (i.e. RStudio) and launch the server from the web. Serena recommended using incognito window b/c OOD usually works better without the caching.</p>
<div class="highlight"><pre><span></span><code>ssh username@login.discovery.neu.edu
</code></pre></div>
<h2 id="server-structure">Server structure</h2>
<p>We have a 30 TB maximum in our working space <code>/work/gmgi</code>:</p>
<ul>
<li>Each lab has their own subfolder that serves as their storage and working space (e.g., <code>work/gmgi/Fisheries/</code>).  </li>
<li><code>databases/</code>: Shared folder for common databases for amplicon sequencing (i.e., 12S, 16S, 18S, COI) and NCBI nt database. View the README file for databases sources.</li>
<li><code>containers/</code>: Shared folder for custom built containers.  </li>
<li><code>packages/</code>: Shared folder for programs downloaded for all users.  </li>
<li><code>miniconda3/</code> and <code>Miniconda3-latest-Linux-x86_64.sh</code>: Shared resource for building conda environments. Environments built here can be used by everyone. <em>Do not edit.</em>  </li>
<li><code>check_storage.sh</code>: Bash script built to calculate TB usage from each lab's folder and gmgi's work and output is <code>storage_summary_2024-09-23.txt</code> with the date calculated.</li>
</ul>
<div class="highlight"><pre><span></span><code>[e.strand@login-00 ~]$ cd /work/gmgi
[e.strand@login-00 gmgi]$ ls
check_storage.sh  containers  databases  ecosystem-diversity  Fisheries  miniconda3  Miniconda3-latest-Linux-x86_64.sh  packages  storage_summary_2024-09-23.txt
</code></pre></div>
<p>Fisheries folder (<code>work/gmgi/Fisheries/</code>) is split by the type of project. <code>reference_genomes</code> includes .fasta reference files for organisms rather than a database (<code>Haddock_ref.fasta</code>). </p>
<div class="highlight"><pre><span></span><code>[e.strand@login-00 Fisheries]$ ls
eDNA  epiage  reference_genomes
</code></pre></div>
<h2 id="storage-rules">Storage rules</h2>
<p>While analyzing data, NOT just at the end of a project!</p>
<p>Raw data files are backed up on AWS services and on GMGI RHEL Gadus immediately upon receiving data. If working on NU cluster, once user is happy with data analysis pipeline, raw and final data is to be removed from NU cluster and only kept on AWS services or GMGI's RHEL server. </p>
<p>Compressing files:  </p>
<ul>
<li>Gzip all fastq files (e.g., raw data, trimmed data), .fasta/.fa files (e.g., reference genomes), and large .txt files (e.g., intermediate files created during analysis): <code>gzip *.fastq</code> or create a slurm array with a sbatch script.    </li>
<li><a href="https://www.genozip.com/standard">Genozip</a> all .bam, .sam, .vcf files (e.g., intermediate files created during analysis). Genozip has been downloaded in /work/gmgi/packages for general use.   </li>
</ul>
<p>Space-related commands:  </p>
<ul>
<li>List all files within a directory and human-readable sizes (folder size is not total size of folder contents): <code>ls -lha</code>  </li>
<li>Calculate total storage taken up by one directory (change path as needed): <code>du -shc .[^.]* /work/gmgi/fisheries</code>    </li>
<li>In /work/gmgi/, there is a <code>check_storage.sh</code> bash script that will use the above commands to create a summary .txt file with the storage use of each team.  </li>
</ul>
<h2 id="nu-contacts-and-research-computing-help">NU Contacts and Research Computing Help</h2>
<p>GMGI's two main contacts are: Greg Shomo (g.shomo@northeastern.edu) and Serena Caplins (s.caplins@northeastern.edu). </p>
<p>If you need assistance, NU's support team is available at rchelp@northeastern.edu or consult the <a href="https://rc-docs.northeastern.edu/en/latest/faq.html#faq">Frequently Asked Questions (FAQs)</a>. Emailing the rchelp team will create a help ticket one of NU's team members will be assigned to your case. When creating a help ticket, CC Geoff Trussell (g.trussell@northeastern.edu) and Jon Grabowski (j.grabowski@northeastern.edu). </p>
<p>RC Help hosts office hours to connect with RC staff and Graduate Research Assistants (GRAs) to ask questions about any RC-related questions that you might have (e.g., Setting up conda environments, Installing software, Optimizing the runtime of your sbatch scripts, Effectively using the Open onDemand website):<br />
- Wednesdays 3 pm - 4 pm: <a href="https://url2.mailanyone.net/scanner?m=1rzNQw-0008Ns-3f&amp;d=4%7Cmail%2F90%2F1713906600%2F1rzNQw-0008Ns-3f%7Cin2e%7C57e1b682%7C28509242%7C14152682%7C6628242AF22ECB27DE9BAA81E1B29580&amp;o=%2Fphto%3A%2Fntseertnstrhaso.zj.u%2Fom20%2F951142466&amp;s=ruDKyxUNCNd62B7o2lO1X9v5I5U">zoom link</a><br />
- Thursdays 11 am - 12 pm: <a href="https://url2.mailanyone.net/scanner?m=1rzNQw-0008Ns-3f&amp;d=4%7Cmail%2F90%2F1713906600%2F1rzNQw-0008Ns-3f%7Cin2e%7C57e1b682%7C28509242%7C14152682%7C6628242AF22ECB27DE9BAA81E1B29580&amp;o=%2Fphto%3A%2Fntseertnstrhaso.zj.u%2Fom63%2F914254083&amp;s=feG6PL708qP4uNB_somz55Vucz0">zoom link</a>  </p>
<p>Ask 2-3 others at GMGI for assistance and/or attend office hours prior to creating a help ticket. We do not want to overwhelm the rchelp desk if we can troubleshoot internally first. GMGI has a #bioinformatics slack channel for this purpose.</p>
<h2 id="running-a-bioinformatic-script">Running a bioinformatic script</h2>
<p>Read through: <a href="https://rc-docs.northeastern.edu/en/latest/runningjobs/index.html">Running Jobs - RC RTD (northeastern.edu)</a> before starting. </p>
<p>Jobs are run either through:<br />
1. Interactive mode (immediate execution and feedback): <code>srun --pty bash</code> to claim a node and then utilize <code>bash scriptname.sh</code> to run a script.  <br />
2. Batch jobs: using scripts to manage longer-running jobs: <code>sbatch scriptname.sh</code> to run a script.  </p>
<p>Interactive mode would be equivalent to running a job directly in your terminal window without starting a tmux session. Using batch jobs and shell scripts would be similar to tmux session where you can turn off wifi, walk away, etc. Interactive requires you to stay connected. </p>
<p>Introduction to Slurm scripts:    </p>
<ul>
<li><a href="https://rc-docs.northeastern.edu/en/latest/slurmguide/index.html">Slurm - RC RTD (northeastern.edu)</a>     </li>
<li><a href="https://rc-docs.northeastern.edu/en/latest/best-practices/index.html">Best Practices - RC RTD (northeastern.edu)</a>    </li>
</ul>
<h2 id="packages-and-modules">Packages and modules</h2>
<p>NU has some modules downloaded that are accessible for all users. Otherwise, larger packages should be installed in a conda environment by the user or simple packages can be downloaded to the <code>/work/gmgi/packages/</code> folder. Instructions for downloading to conda env: [https://rc-docs.northeastern.edu/en/latest/software/index.html]</p>
<p>Common commands:   </p>
<ul>
<li>To find already installed programs: <code>module avail</code>      </li>
<li>To get information about a module: <code>module help [module/version]</code> or <code>module whatis [module/version]</code>. "help" will provide what the module is, package information including version and install date, and a link to the documentation/github. "whatis" will provide a short, one line description of the program.    </li>
<li>To load a module: <code>module load [module/version]</code> (e.g., <code>module load bamUtil/v1.0.15</code>). Loading a module will put all the necessary executables and dependencies for that program in your path so you can call the commands from any location (i.e. your working directory).   </li>
</ul>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": ["content.action.edit", "content.code.copy"], "search": "../assets/javascripts/workers/search.6ce7567c.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.d6f25eb3.min.js"></script>
      
    
  </body>
</html>